// @atproto/api/dist/client/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ComAtprotoServerDescribeServer = exports.ComAtprotoServerDeleteSession = exports.ComAtprotoServerDeleteAccount = exports.ComAtprotoServerDefs = exports.ComAtprotoServerDeactivateAccount = exports.ComAtprotoServerCreateSession = exports.ComAtprotoServerCreateInviteCodes = exports.ComAtprotoServerCreateInviteCode = exports.ComAtprotoServerCreateAppPassword = exports.ComAtprotoServerCreateAccount = exports.ComAtprotoServerConfirmEmail = exports.ComAtprotoServerCheckAccountStatus = exports.ComAtprotoServerActivateAccount = exports.ComAtprotoRepoUploadBlob = exports.ComAtprotoRepoStrongRef = exports.ComAtprotoRepoPutRecord = exports.ComAtprotoRepoListRecords = exports.ComAtprotoRepoListMissingBlobs = exports.ComAtprotoRepoImportRepo = exports.ComAtprotoRepoGetRecord = exports.ComAtprotoRepoDescribeRepo = exports.ComAtprotoRepoDeleteRecord = exports.ComAtprotoRepoCreateRecord = exports.ComAtprotoRepoApplyWrites = exports.ComAtprotoModerationDefs = exports.ComAtprotoModerationCreateReport = exports.ComAtprotoLabelSubscribeLabels = exports.ComAtprotoLabelQueryLabels = exports.ComAtprotoLabelDefs = exports.ComAtprotoIdentityUpdateHandle = exports.ComAtprotoIdentitySubmitPlcOperation = exports.ComAtprotoIdentitySignPlcOperation = exports.ComAtprotoIdentityResolveHandle = exports.ComAtprotoIdentityRequestPlcOperationSignature = exports.ComAtprotoIdentityGetRecommendedDidCredentials = exports.ComAtprotoAdminUpdateSubjectStatus = exports.ComAtprotoAdminUpdateAccountPassword = exports.ComAtprotoAdminUpdateAccountHandle = exports.ComAtprotoAdminUpdateAccountEmail = exports.ComAtprotoAdminSendEmail = exports.ComAtprotoAdminSearchAccounts = exports.ComAtprotoAdminGetSubjectStatus = exports.ComAtprotoAdminGetInviteCodes = exports.ComAtprotoAdminGetAccountInfos = exports.ComAtprotoAdminGetAccountInfo = exports.ComAtprotoAdminEnableAccountInvites = exports.ComAtprotoAdminDisableInviteCodes = exports.ComAtprotoAdminDisableAccountInvites = exports.ComAtprotoAdminDeleteAccount = exports.ComAtprotoAdminDefs = void 0;
exports.AppBskyFeedGetFeedGenerator = exports.AppBskyFeedGetFeed = exports.AppBskyFeedGetAuthorFeed = exports.AppBskyFeedGetActorLikes = exports.AppBskyFeedGetActorFeeds = exports.AppBskyFeedGenerator = exports.AppBskyFeedDescribeFeedGenerator = exports.AppBskyFeedDefs = exports.AppBskyEmbedRecordWithMedia = exports.AppBskyEmbedRecord = exports.AppBskyEmbedImages = exports.AppBskyEmbedExternal = exports.AppBskyActorSearchActorsTypeahead = exports.AppBskyActorSearchActors = exports.AppBskyActorPutPreferences = exports.AppBskyActorProfile = exports.AppBskyActorGetSuggestions = exports.AppBskyActorGetProfiles = exports.AppBskyActorGetProfile = exports.AppBskyActorGetPreferences = exports.AppBskyActorDefs = exports.ComAtprotoTempRequestPhoneVerification = exports.ComAtprotoTempFetchLabels = exports.ComAtprotoTempCheckSignupQueue = exports.ComAtprotoSyncSubscribeRepos = exports.ComAtprotoSyncRequestCrawl = exports.ComAtprotoSyncNotifyOfUpdate = exports.ComAtprotoSyncListRepos = exports.ComAtprotoSyncListBlobs = exports.ComAtprotoSyncGetRepoStatus = exports.ComAtprotoSyncGetRepo = exports.ComAtprotoSyncGetRecord = exports.ComAtprotoSyncGetLatestCommit = exports.ComAtprotoSyncGetHead = exports.ComAtprotoSyncGetCheckout = exports.ComAtprotoSyncGetBlocks = exports.ComAtprotoSyncGetBlob = exports.ComAtprotoServerUpdateEmail = exports.ComAtprotoServerRevokeAppPassword = exports.ComAtprotoServerResetPassword = exports.ComAtprotoServerReserveSigningKey = exports.ComAtprotoServerRequestPasswordReset = exports.ComAtprotoServerRequestEmailUpdate = exports.ComAtprotoServerRequestEmailConfirmation = exports.ComAtprotoServerRequestAccountDelete = exports.ComAtprotoServerRefreshSession = exports.ComAtprotoServerListAppPasswords = exports.ComAtprotoServerGetSession = exports.ComAtprotoServerGetServiceAuth = exports.ComAtprotoServerGetAccountInviteCodes = void 0;
exports.AppBskyNotificationUpdateSeen = exports.AppBskyNotificationRegisterPush = exports.AppBskyNotificationPutPreferences = exports.AppBskyNotificationListNotifications = exports.AppBskyNotificationGetUnreadCount = exports.AppBskyLabelerService = exports.AppBskyLabelerGetServices = exports.AppBskyLabelerDefs = exports.AppBskyGraphUnmuteThread = exports.AppBskyGraphUnmuteActorList = exports.AppBskyGraphUnmuteActor = exports.AppBskyGraphStarterpack = exports.AppBskyGraphMuteThread = exports.AppBskyGraphMuteActorList = exports.AppBskyGraphMuteActor = exports.AppBskyGraphListitem = exports.AppBskyGraphListblock = exports.AppBskyGraphList = exports.AppBskyGraphGetSuggestedFollowsByActor = exports.AppBskyGraphGetStarterPacks = exports.AppBskyGraphGetStarterPack = exports.AppBskyGraphGetRelationships = exports.AppBskyGraphGetMutes = exports.AppBskyGraphGetLists = exports.AppBskyGraphGetListMutes = exports.AppBskyGraphGetListBlocks = exports.AppBskyGraphGetList = exports.AppBskyGraphGetKnownFollowers = exports.AppBskyGraphGetFollows = exports.AppBskyGraphGetFollowers = exports.AppBskyGraphGetBlocks = exports.AppBskyGraphGetActorStarterPacks = exports.AppBskyGraphFollow = exports.AppBskyGraphDefs = exports.AppBskyGraphBlock = exports.AppBskyFeedThreadgate = exports.AppBskyFeedSendInteractions = exports.AppBskyFeedSearchPosts = exports.AppBskyFeedRepost = exports.AppBskyFeedPost = exports.AppBskyFeedLike = exports.AppBskyFeedGetTimeline = exports.AppBskyFeedGetSuggestedFeeds = exports.AppBskyFeedGetRepostedBy = exports.AppBskyFeedGetPosts = exports.AppBskyFeedGetPostThread = exports.AppBskyFeedGetListFeed = exports.AppBskyFeedGetLikes = exports.AppBskyFeedGetFeedSkeleton = exports.AppBskyFeedGetFeedGenerators = void 0;
exports.TOOLS_OZONE_MODERATION = exports.APP_BSKY_GRAPH = exports.APP_BSKY_FEED = exports.COM_ATPROTO_MODERATION = exports.ToolsOzoneTeamUpdateMember = exports.ToolsOzoneTeamListMembers = exports.ToolsOzoneTeamDeleteMember = exports.ToolsOzoneTeamDefs = exports.ToolsOzoneTeamAddMember = exports.ToolsOzoneServerGetConfig = exports.ToolsOzoneModerationSearchRepos = exports.ToolsOzoneModerationQueryStatuses = exports.ToolsOzoneModerationQueryEvents = exports.ToolsOzoneModerationGetRepo = exports.ToolsOzoneModerationGetRecord = exports.ToolsOzoneModerationGetEvent = exports.ToolsOzoneModerationEmitEvent = exports.ToolsOzoneModerationDefs = exports.ToolsOzoneCommunicationUpdateTemplate = exports.ToolsOzoneCommunicationListTemplates = exports.ToolsOzoneCommunicationDeleteTemplate = exports.ToolsOzoneCommunicationDefs = exports.ToolsOzoneCommunicationCreateTemplate = exports.ChatBskyModerationUpdateActorAccess = exports.ChatBskyModerationGetMessageContext = exports.ChatBskyModerationGetActorMetadata = exports.ChatBskyConvoUpdateRead = exports.ChatBskyConvoUnmuteConvo = exports.ChatBskyConvoSendMessageBatch = exports.ChatBskyConvoSendMessage = exports.ChatBskyConvoMuteConvo = exports.ChatBskyConvoListConvos = exports.ChatBskyConvoLeaveConvo = exports.ChatBskyConvoGetMessages = exports.ChatBskyConvoGetLog = exports.ChatBskyConvoGetConvoForMembers = exports.ChatBskyConvoGetConvo = exports.ChatBskyConvoDeleteMessageForSelf = exports.ChatBskyConvoDefs = exports.ChatBskyActorExportAccountData = exports.ChatBskyActorDeleteAccount = exports.ChatBskyActorDefs = exports.ChatBskyActorDeclaration = exports.AppBskyUnspeccedSearchPostsSkeleton = exports.AppBskyUnspeccedSearchActorsSkeleton = exports.AppBskyUnspeccedGetTaggedSuggestions = exports.AppBskyUnspeccedGetSuggestionsSkeleton = exports.AppBskyUnspeccedGetPopularFeedGenerators = exports.AppBskyUnspeccedDefs = exports.AppBskyRichtextFacet = void 0;
exports.ToolsOzoneTeamNS = exports.ToolsOzoneServerNS = exports.ToolsOzoneModerationNS = exports.ToolsOzoneCommunicationNS = exports.ToolsOzoneNS = exports.ToolsNS = exports.ChatBskyModerationNS = exports.ChatBskyConvoNS = exports.DeclarationRecord = exports.ChatBskyActorNS = exports.ChatBskyNS = exports.ChatNS = exports.AppBskyUnspeccedNS = exports.AppBskyRichtextNS = exports.AppBskyNotificationNS = exports.ServiceRecord = exports.AppBskyLabelerNS = exports.StarterpackRecord = exports.ListitemRecord = exports.ListblockRecord = exports.ListRecord = exports.FollowRecord = exports.BlockRecord = exports.AppBskyGraphNS = exports.ThreadgateRecord = exports.RepostRecord = exports.PostRecord = exports.LikeRecord = exports.GeneratorRecord = exports.AppBskyFeedNS = exports.AppBskyEmbedNS = exports.ProfileRecord = exports.AppBskyActorNS = exports.AppBskyNS = exports.AppNS = exports.ComAtprotoTempNS = exports.ComAtprotoSyncNS = exports.ComAtprotoServerNS = exports.ComAtprotoRepoNS = exports.ComAtprotoModerationNS = exports.ComAtprotoLabelNS = exports.ComAtprotoIdentityNS = exports.ComAtprotoAdminNS = exports.ComAtprotoNS = exports.ComNS = exports.AtpServiceClient = exports.AtpBaseClient = exports.TOOLS_OZONE_TEAM = void 0;
/**
 * GENERATED CODE - DO NOT MODIFY
 */
const xrpc_1 = require("@atproto/xrpc");
const lexicons_1 = require("./lexicons");
const ComAtprotoAdminDeleteAccount = __importStar(require("./types/com/atproto/admin/deleteAccount"));
const ComAtprotoAdminDisableAccountInvites = __importStar(require("./types/com/atproto/admin/disableAccountInvites"));
const ComAtprotoAdminDisableInviteCodes = __importStar(require("./types/com/atproto/admin/disableInviteCodes"));
const ComAtprotoAdminEnableAccountInvites = __importStar(require("./types/com/atproto/admin/enableAccountInvites"));
const ComAtprotoAdminGetAccountInfo = __importStar(require("./types/com/atproto/admin/getAccountInfo"));
const ComAtprotoAdminGetAccountInfos = __importStar(require("./types/com/atproto/admin/getAccountInfos"));
const ComAtprotoAdminGetInviteCodes = __importStar(require("./types/com/atproto/admin/getInviteCodes"));
const ComAtprotoAdminGetSubjectStatus = __importStar(require("./types/com/atproto/admin/getSubjectStatus"));
const ComAtprotoAdminSearchAccounts = __importStar(require("./types/com/atproto/admin/searchAccounts"));
const ComAtprotoAdminSendEmail = __importStar(require("./types/com/atproto/admin/sendEmail"));
const ComAtprotoAdminUpdateAccountEmail = __importStar(require("./types/com/atproto/admin/updateAccountEmail"));
const ComAtprotoAdminUpdateAccountHandle = __importStar(require("./types/com/atproto/admin/updateAccountHandle"));
const ComAtprotoAdminUpdateAccountPassword = __importStar(require("./types/com/atproto/admin/updateAccountPassword"));
const ComAtprotoAdminUpdateSubjectStatus = __importStar(require("./types/com/atproto/admin/updateSubjectStatus"));
const ComAtprotoIdentityGetRecommendedDidCredentials = __importStar(require("./types/com/atproto/identity/getRecommendedDidCredentials"));
const ComAtprotoIdentityRequestPlcOperationSignature = __importStar(require("./types/com/atproto/identity/requestPlcOperationSignature"));
const ComAtprotoIdentityResolveHandle = __importStar(require("./types/com/atproto/identity/resolveHandle"));
const ComAtprotoIdentitySignPlcOperation = __importStar(require("./types/com/atproto/identity/signPlcOperation"));
const ComAtprotoIdentitySubmitPlcOperation = __importStar(require("./types/com/atproto/identity/submitPlcOperation"));
const ComAtprotoIdentityUpdateHandle = __importStar(require("./types/com/atproto/identity/updateHandle"));
const ComAtprotoLabelQueryLabels = __importStar(require("./types/com/atproto/label/queryLabels"));
const ComAtprotoModerationCreateReport = __importStar(require("./types/com/atproto/moderation/createReport"));
const ComAtprotoRepoApplyWrites = __importStar(require("./types/com/atproto/repo/applyWrites"));
const ComAtprotoRepoCreateRecord = __importStar(require("./types/com/atproto/repo/createRecord"));
const ComAtprotoRepoDeleteRecord = __importStar(require("./types/com/atproto/repo/deleteRecord"));
const ComAtprotoRepoDescribeRepo = __importStar(require("./types/com/atproto/repo/describeRepo"));
const ComAtprotoRepoGetRecord = __importStar(require("./types/com/atproto/repo/getRecord"));
const ComAtprotoRepoImportRepo = __importStar(require("./types/com/atproto/repo/importRepo"));
const ComAtprotoRepoListMissingBlobs = __importStar(require("./types/com/atproto/repo/listMissingBlobs"));
const ComAtprotoRepoListRecords = __importStar(require("./types/com/atproto/repo/listRecords"));
const ComAtprotoRepoPutRecord = __importStar(require("./types/com/atproto/repo/putRecord"));
const ComAtprotoRepoUploadBlob = __importStar(require("./types/com/atproto/repo/uploadBlob"));
const ComAtprotoServerActivateAccount = __importStar(require("./types/com/atproto/server/activateAccount"));
const ComAtprotoServerCheckAccountStatus = __importStar(require("./types/com/atproto/server/checkAccountStatus"));
const ComAtprotoServerConfirmEmail = __importStar(require("./types/com/atproto/server/confirmEmail"));
const ComAtprotoServerCreateAccount = __importStar(require("./types/com/atproto/server/createAccount"));
const ComAtprotoServerCreateAppPassword = __importStar(require("./types/com/atproto/server/createAppPassword"));
const ComAtprotoServerCreateInviteCode = __importStar(require("./types/com/atproto/server/createInviteCode"));
const ComAtprotoServerCreateInviteCodes = __importStar(require("./types/com/atproto/server/createInviteCodes"));
const ComAtprotoServerCreateSession = __importStar(require("./types/com/atproto/server/createSession"));
const ComAtprotoServerDeactivateAccount = __importStar(require("./types/com/atproto/server/deactivateAccount"));
const ComAtprotoServerDeleteAccount = __importStar(require("./types/com/atproto/server/deleteAccount"));
const ComAtprotoServerDeleteSession = __importStar(require("./types/com/atproto/server/deleteSession"));
const ComAtprotoServerDescribeServer = __importStar(require("./types/com/atproto/server/describeServer"));
const ComAtprotoServerGetAccountInviteCodes = __importStar(require("./types/com/atproto/server/getAccountInviteCodes"));
const ComAtprotoServerGetServiceAuth = __importStar(require("./types/com/atproto/server/getServiceAuth"));
const ComAtprotoServerGetSession = __importStar(require("./types/com/atproto/server/getSession"));
const ComAtprotoServerListAppPasswords = __importStar(require("./types/com/atproto/server/listAppPasswords"));
const ComAtprotoServerRefreshSession = __importStar(require("./types/com/atproto/server/refreshSession"));
const ComAtprotoServerRequestAccountDelete = __importStar(require("./types/com/atproto/server/requestAccountDelete"));
const ComAtprotoServerRequestEmailConfirmation = __importStar(require("./types/com/atproto/server/requestEmailConfirmation"));
const ComAtprotoServerRequestEmailUpdate = __importStar(require("./types/com/atproto/server/requestEmailUpdate"));
const ComAtprotoServerRequestPasswordReset = __importStar(require("./types/com/atproto/server/requestPasswordReset"));
const ComAtprotoServerReserveSigningKey = __importStar(require("./types/com/atproto/server/reserveSigningKey"));
const ComAtprotoServerResetPassword = __importStar(require("./types/com/atproto/server/resetPassword"));
const ComAtprotoServerRevokeAppPassword = __importStar(require("./types/com/atproto/server/revokeAppPassword"));
const ComAtprotoServerUpdateEmail = __importStar(require("./types/com/atproto/server/updateEmail"));
const ComAtprotoSyncGetBlob = __importStar(require("./types/com/atproto/sync/getBlob"));
const ComAtprotoSyncGetBlocks = __importStar(require("./types/com/atproto/sync/getBlocks"));
const ComAtprotoSyncGetCheckout = __importStar(require("./types/com/atproto/sync/getCheckout"));
const ComAtprotoSyncGetHead = __importStar(require("./types/com/atproto/sync/getHead"));
const ComAtprotoSyncGetLatestCommit = __importStar(require("./types/com/atproto/sync/getLatestCommit"));
const ComAtprotoSyncGetRecord = __importStar(require("./types/com/atproto/sync/getRecord"));
const ComAtprotoSyncGetRepo = __importStar(require("./types/com/atproto/sync/getRepo"));
const ComAtprotoSyncGetRepoStatus = __importStar(require("./types/com/atproto/sync/getRepoStatus"));
const ComAtprotoSyncListBlobs = __importStar(require("./types/com/atproto/sync/listBlobs"));
const ComAtprotoSyncListRepos = __importStar(require("./types/com/atproto/sync/listRepos"));
const ComAtprotoSyncNotifyOfUpdate = __importStar(require("./types/com/atproto/sync/notifyOfUpdate"));
const ComAtprotoSyncRequestCrawl = __importStar(require("./types/com/atproto/sync/requestCrawl"));
const ComAtprotoTempCheckSignupQueue = __importStar(require("./types/com/atproto/temp/checkSignupQueue"));
const ComAtprotoTempFetchLabels = __importStar(require("./types/com/atproto/temp/fetchLabels"));
const ComAtprotoTempRequestPhoneVerification = __importStar(require("./types/com/atproto/temp/requestPhoneVerification"));
const AppBskyActorGetPreferences = __importStar(require("./types/app/bsky/actor/getPreferences"));
const AppBskyActorGetProfile = __importStar(require("./types/app/bsky/actor/getProfile"));
const AppBskyActorGetProfiles = __importStar(require("./types/app/bsky/actor/getProfiles"));
const AppBskyActorGetSuggestions = __importStar(require("./types/app/bsky/actor/getSuggestions"));
const AppBskyActorPutPreferences = __importStar(require("./types/app/bsky/actor/putPreferences"));
const AppBskyActorSearchActors = __importStar(require("./types/app/bsky/actor/searchActors"));
const AppBskyActorSearchActorsTypeahead = __importStar(require("./types/app/bsky/actor/searchActorsTypeahead"));
const AppBskyFeedDescribeFeedGenerator = __importStar(require("./types/app/bsky/feed/describeFeedGenerator"));
const AppBskyFeedGetActorFeeds = __importStar(require("./types/app/bsky/feed/getActorFeeds"));
const AppBskyFeedGetActorLikes = __importStar(require("./types/app/bsky/feed/getActorLikes"));
const AppBskyFeedGetAuthorFeed = __importStar(require("./types/app/bsky/feed/getAuthorFeed"));
const AppBskyFeedGetFeed = __importStar(require("./types/app/bsky/feed/getFeed"));
const AppBskyFeedGetFeedGenerator = __importStar(require("./types/app/bsky/feed/getFeedGenerator"));
const AppBskyFeedGetFeedGenerators = __importStar(require("./types/app/bsky/feed/getFeedGenerators"));
const AppBskyFeedGetFeedSkeleton = __importStar(require("./types/app/bsky/feed/getFeedSkeleton"));
const AppBskyFeedGetLikes = __importStar(require("./types/app/bsky/feed/getLikes"));
const AppBskyFeedGetListFeed = __importStar(require("./types/app/bsky/feed/getListFeed"));
const AppBskyFeedGetPostThread = __importStar(require("./types/app/bsky/feed/getPostThread"));
const AppBskyFeedGetPosts = __importStar(require("./types/app/bsky/feed/getPosts"));
const AppBskyFeedGetRepostedBy = __importStar(require("./types/app/bsky/feed/getRepostedBy"));
const AppBskyFeedGetSuggestedFeeds = __importStar(require("./types/app/bsky/feed/getSuggestedFeeds"));
const AppBskyFeedGetTimeline = __importStar(require("./types/app/bsky/feed/getTimeline"));
const AppBskyFeedSearchPosts = __importStar(require("./types/app/bsky/feed/searchPosts"));
const AppBskyFeedSendInteractions = __importStar(require("./types/app/bsky/feed/sendInteractions"));
const AppBskyGraphGetActorStarterPacks = __importStar(require("./types/app/bsky/graph/getActorStarterPacks"));
const AppBskyGraphGetBlocks = __importStar(require("./types/app/bsky/graph/getBlocks"));
const AppBskyGraphGetFollowers = __importStar(require("./types/app/bsky/graph/getFollowers"));
const AppBskyGraphGetFollows = __importStar(require("./types/app/bsky/graph/getFollows"));
const AppBskyGraphGetKnownFollowers = __importStar(require("./types/app/bsky/graph/getKnownFollowers"));
const AppBskyGraphGetList = __importStar(require("./types/app/bsky/graph/getList"));
const AppBskyGraphGetListBlocks = __importStar(require("./types/app/bsky/graph/getListBlocks"));
const AppBskyGraphGetListMutes = __importStar(require("./types/app/bsky/graph/getListMutes"));
const AppBskyGraphGetLists = __importStar(require("./types/app/bsky/graph/getLists"));
const AppBskyGraphGetMutes = __importStar(require("./types/app/bsky/graph/getMutes"));
const AppBskyGraphGetRelationships = __importStar(require("./types/app/bsky/graph/getRelationships"));
const AppBskyGraphGetStarterPack = __importStar(require("./types/app/bsky/graph/getStarterPack"));
const AppBskyGraphGetStarterPacks = __importStar(require("./types/app/bsky/graph/getStarterPacks"));
const AppBskyGraphGetSuggestedFollowsByActor = __importStar(require("./types/app/bsky/graph/getSuggestedFollowsByActor"));
const AppBskyGraphMuteActor = __importStar(require("./types/app/bsky/graph/muteActor"));
const AppBskyGraphMuteActorList = __importStar(require("./types/app/bsky/graph/muteActorList"));
const AppBskyGraphMuteThread = __importStar(require("./types/app/bsky/graph/muteThread"));
const AppBskyGraphUnmuteActor = __importStar(require("./types/app/bsky/graph/unmuteActor"));
const AppBskyGraphUnmuteActorList = __importStar(require("./types/app/bsky/graph/unmuteActorList"));
const AppBskyGraphUnmuteThread = __importStar(require("./types/app/bsky/graph/unmuteThread"));
const AppBskyLabelerGetServices = __importStar(require("./types/app/bsky/labeler/getServices"));
const AppBskyNotificationGetUnreadCount = __importStar(require("./types/app/bsky/notification/getUnreadCount"));
const AppBskyNotificationListNotifications = __importStar(require("./types/app/bsky/notification/listNotifications"));
const AppBskyNotificationPutPreferences = __importStar(require("./types/app/bsky/notification/putPreferences"));
const AppBskyNotificationRegisterPush = __importStar(require("./types/app/bsky/notification/registerPush"));
const AppBskyNotificationUpdateSeen = __importStar(require("./types/app/bsky/notification/updateSeen"));
const AppBskyUnspeccedGetPopularFeedGenerators = __importStar(require("./types/app/bsky/unspecced/getPopularFeedGenerators"));
const AppBskyUnspeccedGetSuggestionsSkeleton = __importStar(require("./types/app/bsky/unspecced/getSuggestionsSkeleton"));
const AppBskyUnspeccedGetTaggedSuggestions = __importStar(require("./types/app/bsky/unspecced/getTaggedSuggestions"));
const AppBskyUnspeccedSearchActorsSkeleton = __importStar(require("./types/app/bsky/unspecced/searchActorsSkeleton"));
const AppBskyUnspeccedSearchPostsSkeleton = __importStar(require("./types/app/bsky/unspecced/searchPostsSkeleton"));
const ChatBskyActorDeleteAccount = __importStar(require("./types/chat/bsky/actor/deleteAccount"));
const ChatBskyActorExportAccountData = __importStar(require("./types/chat/bsky/actor/exportAccountData"));
const ChatBskyConvoDeleteMessageForSelf = __importStar(require("./types/chat/bsky/convo/deleteMessageForSelf"));
const ChatBskyConvoGetConvo = __importStar(require("./types/chat/bsky/convo/getConvo"));
const ChatBskyConvoGetConvoForMembers = __importStar(require("./types/chat/bsky/convo/getConvoForMembers"));
const ChatBskyConvoGetLog = __importStar(require("./types/chat/bsky/convo/getLog"));
const ChatBskyConvoGetMessages = __importStar(require("./types/chat/bsky/convo/getMessages"));
const ChatBskyConvoLeaveConvo = __importStar(require("./types/chat/bsky/convo/leaveConvo"));
const ChatBskyConvoListConvos = __importStar(require("./types/chat/bsky/convo/listConvos"));
const ChatBskyConvoMuteConvo = __importStar(require("./types/chat/bsky/convo/muteConvo"));
const ChatBskyConvoSendMessage = __importStar(require("./types/chat/bsky/convo/sendMessage"));
const ChatBskyConvoSendMessageBatch = __importStar(require("./types/chat/bsky/convo/sendMessageBatch"));
const ChatBskyConvoUnmuteConvo = __importStar(require("./types/chat/bsky/convo/unmuteConvo"));
const ChatBskyConvoUpdateRead = __importStar(require("./types/chat/bsky/convo/updateRead"));
const ChatBskyModerationGetActorMetadata = __importStar(require("./types/chat/bsky/moderation/getActorMetadata"));
const ChatBskyModerationGetMessageContext = __importStar(require("./types/chat/bsky/moderation/getMessageContext"));
const ChatBskyModerationUpdateActorAccess = __importStar(require("./types/chat/bsky/moderation/updateActorAccess"));
const ToolsOzoneCommunicationCreateTemplate = __importStar(require("./types/tools/ozone/communication/createTemplate"));
const ToolsOzoneCommunicationDeleteTemplate = __importStar(require("./types/tools/ozone/communication/deleteTemplate"));
const ToolsOzoneCommunicationListTemplates = __importStar(require("./types/tools/ozone/communication/listTemplates"));
const ToolsOzoneCommunicationUpdateTemplate = __importStar(require("./types/tools/ozone/communication/updateTemplate"));
const ToolsOzoneModerationEmitEvent = __importStar(require("./types/tools/ozone/moderation/emitEvent"));
const ToolsOzoneModerationGetEvent = __importStar(require("./types/tools/ozone/moderation/getEvent"));
const ToolsOzoneModerationGetRecord = __importStar(require("./types/tools/ozone/moderation/getRecord"));
const ToolsOzoneModerationGetRepo = __importStar(require("./types/tools/ozone/moderation/getRepo"));
const ToolsOzoneModerationQueryEvents = __importStar(require("./types/tools/ozone/moderation/queryEvents"));
const ToolsOzoneModerationQueryStatuses = __importStar(require("./types/tools/ozone/moderation/queryStatuses"));
const ToolsOzoneModerationSearchRepos = __importStar(require("./types/tools/ozone/moderation/searchRepos"));
const ToolsOzoneServerGetConfig = __importStar(require("./types/tools/ozone/server/getConfig"));
const ToolsOzoneTeamAddMember = __importStar(require("./types/tools/ozone/team/addMember"));
const ToolsOzoneTeamDeleteMember = __importStar(require("./types/tools/ozone/team/deleteMember"));
const ToolsOzoneTeamListMembers = __importStar(require("./types/tools/ozone/team/listMembers"));
const ToolsOzoneTeamUpdateMember = __importStar(require("./types/tools/ozone/team/updateMember"));
exports.ComAtprotoAdminDefs = __importStar(require("./types/com/atproto/admin/defs"));
exports.ComAtprotoAdminDeleteAccount = __importStar(require("./types/com/atproto/admin/deleteAccount"));
exports.ComAtprotoAdminDisableAccountInvites = __importStar(require("./types/com/atproto/admin/disableAccountInvites"));
exports.ComAtprotoAdminDisableInviteCodes = __importStar(require("./types/com/atproto/admin/disableInviteCodes"));
exports.ComAtprotoAdminEnableAccountInvites = __importStar(require("./types/com/atproto/admin/enableAccountInvites"));
exports.ComAtprotoAdminGetAccountInfo = __importStar(require("./types/com/atproto/admin/getAccountInfo"));
exports.ComAtprotoAdminGetAccountInfos = __importStar(require("./types/com/atproto/admin/getAccountInfos"));
exports.ComAtprotoAdminGetInviteCodes = __importStar(require("./types/com/atproto/admin/getInviteCodes"));
exports.ComAtprotoAdminGetSubjectStatus = __importStar(require("./types/com/atproto/admin/getSubjectStatus"));
exports.ComAtprotoAdminSearchAccounts = __importStar(require("./types/com/atproto/admin/searchAccounts"));
exports.ComAtprotoAdminSendEmail = __importStar(require("./types/com/atproto/admin/sendEmail"));
exports.ComAtprotoAdminUpdateAccountEmail = __importStar(require("./types/com/atproto/admin/updateAccountEmail"));
exports.ComAtprotoAdminUpdateAccountHandle = __importStar(require("./types/com/atproto/admin/updateAccountHandle"));
exports.ComAtprotoAdminUpdateAccountPassword = __importStar(require("./types/com/atproto/admin/updateAccountPassword"));
exports.ComAtprotoAdminUpdateSubjectStatus = __importStar(require("./types/com/atproto/admin/updateSubjectStatus"));
exports.ComAtprotoIdentityGetRecommendedDidCredentials = __importStar(require("./types/com/atproto/identity/getRecommendedDidCredentials"));
exports.ComAtprotoIdentityRequestPlcOperationSignature = __importStar(require("./types/com/atproto/identity/requestPlcOperationSignature"));
exports.ComAtprotoIdentityResolveHandle = __importStar(require("./types/com/atproto/identity/resolveHandle"));
exports.ComAtprotoIdentitySignPlcOperation = __importStar(require("./types/com/atproto/identity/signPlcOperation"));
exports.ComAtprotoIdentitySubmitPlcOperation = __importStar(require("./types/com/atproto/identity/submitPlcOperation"));
exports.ComAtprotoIdentityUpdateHandle = __importStar(require("./types/com/atproto/identity/updateHandle"));
exports.ComAtprotoLabelDefs = __importStar(require("./types/com/atproto/label/defs"));
exports.ComAtprotoLabelQueryLabels = __importStar(require("./types/com/atproto/label/queryLabels"));
exports.ComAtprotoLabelSubscribeLabels = __importStar(require("./types/com/atproto/label/subscribeLabels"));
exports.ComAtprotoModerationCreateReport = __importStar(require("./types/com/atproto/moderation/createReport"));
exports.ComAtprotoModerationDefs = __importStar(require("./types/com/atproto/moderation/defs"));
exports.ComAtprotoRepoApplyWrites = __importStar(require("./types/com/atproto/repo/applyWrites"));
exports.ComAtprotoRepoCreateRecord = __importStar(require("./types/com/atproto/repo/createRecord"));
exports.ComAtprotoRepoDeleteRecord = __importStar(require("./types/com/atproto/repo/deleteRecord"));
exports.ComAtprotoRepoDescribeRepo = __importStar(require("./types/com/atproto/repo/describeRepo"));
exports.ComAtprotoRepoGetRecord = __importStar(require("./types/com/atproto/repo/getRecord"));
exports.ComAtprotoRepoImportRepo = __importStar(require("./types/com/atproto/repo/importRepo"));
exports.ComAtprotoRepoListMissingBlobs = __importStar(require("./types/com/atproto/repo/listMissingBlobs"));
exports.ComAtprotoRepoListRecords = __importStar(require("./types/com/atproto/repo/listRecords"));
exports.ComAtprotoRepoPutRecord = __importStar(require("./types/com/atproto/repo/putRecord"));
exports.ComAtprotoRepoStrongRef = __importStar(require("./types/com/atproto/repo/strongRef"));
exports.ComAtprotoRepoUploadBlob = __importStar(require("./types/com/atproto/repo/uploadBlob"));
exports.ComAtprotoServerActivateAccount = __importStar(require("./types/com/atproto/server/activateAccount"));
exports.ComAtprotoServerCheckAccountStatus = __importStar(require("./types/com/atproto/server/checkAccountStatus"));
exports.ComAtprotoServerConfirmEmail = __importStar(require("./types/com/atproto/server/confirmEmail"));
exports.ComAtprotoServerCreateAccount = __importStar(require("./types/com/atproto/server/createAccount"));
exports.ComAtprotoServerCreateAppPassword = __importStar(require("./types/com/atproto/server/createAppPassword"));
exports.ComAtprotoServerCreateInviteCode = __importStar(require("./types/com/atproto/server/createInviteCode"));
exports.ComAtprotoServerCreateInviteCodes = __importStar(require("./types/com/atproto/server/createInviteCodes"));
exports.ComAtprotoServerCreateSession = __importStar(require("./types/com/atproto/server/createSession"));
exports.ComAtprotoServerDeactivateAccount = __importStar(require("./types/com/atproto/server/deactivateAccount"));
exports.ComAtprotoServerDefs = __importStar(require("./types/com/atproto/server/defs"));
exports.ComAtprotoServerDeleteAccount = __importStar(require("./types/com/atproto/server/deleteAccount"));
exports.ComAtprotoServerDeleteSession = __importStar(require("./types/com/atproto/server/deleteSession"));
exports.ComAtprotoServerDescribeServer = __importStar(require("./types/com/atproto/server/describeServer"));
exports.ComAtprotoServerGetAccountInviteCodes = __importStar(require("./types/com/atproto/server/getAccountInviteCodes"));
exports.ComAtprotoServerGetServiceAuth = __importStar(require("./types/com/atproto/server/getServiceAuth"));
exports.ComAtprotoServerGetSession = __importStar(require("./types/com/atproto/server/getSession"));
exports.ComAtprotoServerListAppPasswords = __importStar(require("./types/com/atproto/server/listAppPasswords"));
exports.ComAtprotoServerRefreshSession = __importStar(require("./types/com/atproto/server/refreshSession"));
exports.ComAtprotoServerRequestAccountDelete = __importStar(require("./types/com/atproto/server/requestAccountDelete"));
exports.ComAtprotoServerRequestEmailConfirmation = __importStar(require("./types/com/atproto/server/requestEmailConfirmation"));
exports.ComAtprotoServerRequestEmailUpdate = __importStar(require("./types/com/atproto/server/requestEmailUpdate"));
exports.ComAtprotoServerRequestPasswordReset = __importStar(require("./types/com/atproto/server/requestPasswordReset"));
exports.ComAtprotoServerReserveSigningKey = __importStar(require("./types/com/atproto/server/reserveSigningKey"));
exports.ComAtprotoServerResetPassword = __importStar(require("./types/com/atproto/server/resetPassword"));
exports.ComAtprotoServerRevokeAppPassword = __importStar(require("./types/com/atproto/server/revokeAppPassword"));
exports.ComAtprotoServerUpdateEmail = __importStar(require("./types/com/atproto/server/updateEmail"));
exports.ComAtprotoSyncGetBlob = __importStar(require("./types/com/atproto/sync/getBlob"));
exports.ComAtprotoSyncGetBlocks = __importStar(require("./types/com/atproto/sync/getBlocks"));
exports.ComAtprotoSyncGetCheckout = __importStar(require("./types/com/atproto/sync/getCheckout"));
exports.ComAtprotoSyncGetHead = __importStar(require("./types/com/atproto/sync/getHead"));
exports.ComAtprotoSyncGetLatestCommit = __importStar(require("./types/com/atproto/sync/getLatestCommit"));
exports.ComAtprotoSyncGetRecord = __importStar(require("./types/com/atproto/sync/getRecord"));
exports.ComAtprotoSyncGetRepo = __importStar(require("./types/com/atproto/sync/getRepo"));
exports.ComAtprotoSyncGetRepoStatus = __importStar(require("./types/com/atproto/sync/getRepoStatus"));
exports.ComAtprotoSyncListBlobs = __importStar(require("./types/com/atproto/sync/listBlobs"));
exports.ComAtprotoSyncListRepos = __importStar(require("./types/com/atproto/sync/listRepos"));
exports.ComAtprotoSyncNotifyOfUpdate = __importStar(require("./types/com/atproto/sync/notifyOfUpdate"));
exports.ComAtprotoSyncRequestCrawl = __importStar(require("./types/com/atproto/sync/requestCrawl"));
exports.ComAtprotoSyncSubscribeRepos = __importStar(require("./types/com/atproto/sync/subscribeRepos"));
exports.ComAtprotoTempCheckSignupQueue = __importStar(require("./types/com/atproto/temp/checkSignupQueue"));
exports.ComAtprotoTempFetchLabels = __importStar(require("./types/com/atproto/temp/fetchLabels"));
exports.ComAtprotoTempRequestPhoneVerification = __importStar(require("./types/com/atproto/temp/requestPhoneVerification"));
exports.AppBskyActorDefs = __importStar(require("./types/app/bsky/actor/defs"));
exports.AppBskyActorGetPreferences = __importStar(require("./types/app/bsky/actor/getPreferences"));
exports.AppBskyActorGetProfile = __importStar(require("./types/app/bsky/actor/getProfile"));
exports.AppBskyActorGetProfiles = __importStar(require("./types/app/bsky/actor/getProfiles"));
exports.AppBskyActorGetSuggestions = __importStar(require("./types/app/bsky/actor/getSuggestions"));
exports.AppBskyActorProfile = __importStar(require("./types/app/bsky/actor/profile"));
exports.AppBskyActorPutPreferences = __importStar(require("./types/app/bsky/actor/putPreferences"));
exports.AppBskyActorSearchActors = __importStar(require("./types/app/bsky/actor/searchActors"));
exports.AppBskyActorSearchActorsTypeahead = __importStar(require("./types/app/bsky/actor/searchActorsTypeahead"));
exports.AppBskyEmbedExternal = __importStar(require("./types/app/bsky/embed/external"));
exports.AppBskyEmbedImages = __importStar(require("./types/app/bsky/embed/images"));
exports.AppBskyEmbedRecord = __importStar(require("./types/app/bsky/embed/record"));
exports.AppBskyEmbedRecordWithMedia = __importStar(require("./types/app/bsky/embed/recordWithMedia"));
exports.AppBskyFeedDefs = __importStar(require("./types/app/bsky/feed/defs"));
exports.AppBskyFeedDescribeFeedGenerator = __importStar(require("./types/app/bsky/feed/describeFeedGenerator"));
exports.AppBskyFeedGenerator = __importStar(require("./types/app/bsky/feed/generator"));
exports.AppBskyFeedGetActorFeeds = __importStar(require("./types/app/bsky/feed/getActorFeeds"));
exports.AppBskyFeedGetActorLikes = __importStar(require("./types/app/bsky/feed/getActorLikes"));
exports.AppBskyFeedGetAuthorFeed = __importStar(require("./types/app/bsky/feed/getAuthorFeed"));
exports.AppBskyFeedGetFeed = __importStar(require("./types/app/bsky/feed/getFeed"));
exports.AppBskyFeedGetFeedGenerator = __importStar(require("./types/app/bsky/feed/getFeedGenerator"));
exports.AppBskyFeedGetFeedGenerators = __importStar(require("./types/app/bsky/feed/getFeedGenerators"));
exports.AppBskyFeedGetFeedSkeleton = __importStar(require("./types/app/bsky/feed/getFeedSkeleton"));
exports.AppBskyFeedGetLikes = __importStar(require("./types/app/bsky/feed/getLikes"));
exports.AppBskyFeedGetListFeed = __importStar(require("./types/app/bsky/feed/getListFeed"));
exports.AppBskyFeedGetPostThread = __importStar(require("./types/app/bsky/feed/getPostThread"));
exports.AppBskyFeedGetPosts = __importStar(require("./types/app/bsky/feed/getPosts"));
exports.AppBskyFeedGetRepostedBy = __importStar(require("./types/app/bsky/feed/getRepostedBy"));
exports.AppBskyFeedGetSuggestedFeeds = __importStar(require("./types/app/bsky/feed/getSuggestedFeeds"));
exports.AppBskyFeedGetTimeline = __importStar(require("./types/app/bsky/feed/getTimeline"));
exports.AppBskyFeedLike = __importStar(require("./types/app/bsky/feed/like"));
exports.AppBskyFeedPost = __importStar(require("./types/app/bsky/feed/post"));
exports.AppBskyFeedRepost = __importStar(require("./types/app/bsky/feed/repost"));
exports.AppBskyFeedSearchPosts = __importStar(require("./types/app/bsky/feed/searchPosts"));
exports.AppBskyFeedSendInteractions = __importStar(require("./types/app/bsky/feed/sendInteractions"));
exports.AppBskyFeedThreadgate = __importStar(require("./types/app/bsky/feed/threadgate"));
exports.AppBskyGraphBlock = __importStar(require("./types/app/bsky/graph/block"));
exports.AppBskyGraphDefs = __importStar(require("./types/app/bsky/graph/defs"));
exports.AppBskyGraphFollow = __importStar(require("./types/app/bsky/graph/follow"));
exports.AppBskyGraphGetActorStarterPacks = __importStar(require("./types/app/bsky/graph/getActorStarterPacks"));
exports.AppBskyGraphGetBlocks = __importStar(require("./types/app/bsky/graph/getBlocks"));
exports.AppBskyGraphGetFollowers = __importStar(require("./types/app/bsky/graph/getFollowers"));
exports.AppBskyGraphGetFollows = __importStar(require("./types/app/bsky/graph/getFollows"));
exports.AppBskyGraphGetKnownFollowers = __importStar(require("./types/app/bsky/graph/getKnownFollowers"));
exports.AppBskyGraphGetList = __importStar(require("./types/app/bsky/graph/getList"));
exports.AppBskyGraphGetListBlocks = __importStar(require("./types/app/bsky/graph/getListBlocks"));
exports.AppBskyGraphGetListMutes = __importStar(require("./types/app/bsky/graph/getListMutes"));
exports.AppBskyGraphGetLists = __importStar(require("./types/app/bsky/graph/getLists"));
exports.AppBskyGraphGetMutes = __importStar(require("./types/app/bsky/graph/getMutes"));
exports.AppBskyGraphGetRelationships = __importStar(require("./types/app/bsky/graph/getRelationships"));
exports.AppBskyGraphGetStarterPack = __importStar(require("./types/app/bsky/graph/getStarterPack"));
exports.AppBskyGraphGetStarterPacks = __importStar(require("./types/app/bsky/graph/getStarterPacks"));
exports.AppBskyGraphGetSuggestedFollowsByActor = __importStar(require("./types/app/bsky/graph/getSuggestedFollowsByActor"));
exports.AppBskyGraphList = __importStar(require("./types/app/bsky/graph/list"));
exports.AppBskyGraphListblock = __importStar(require("./types/app/bsky/graph/listblock"));
exports.AppBskyGraphListitem = __importStar(require("./types/app/bsky/graph/listitem"));
exports.AppBskyGraphMuteActor = __importStar(require("./types/app/bsky/graph/muteActor"));
exports.AppBskyGraphMuteActorList = __importStar(require("./types/app/bsky/graph/muteActorList"));
exports.AppBskyGraphMuteThread = __importStar(require("./types/app/bsky/graph/muteThread"));
exports.AppBskyGraphStarterpack = __importStar(require("./types/app/bsky/graph/starterpack"));
exports.AppBskyGraphUnmuteActor = __importStar(require("./types/app/bsky/graph/unmuteActor"));
exports.AppBskyGraphUnmuteActorList = __importStar(require("./types/app/bsky/graph/unmuteActorList"));
exports.AppBskyGraphUnmuteThread = __importStar(require("./types/app/bsky/graph/unmuteThread"));
exports.AppBskyLabelerDefs = __importStar(require("./types/app/bsky/labeler/defs"));
exports.AppBskyLabelerGetServices = __importStar(require("./types/app/bsky/labeler/getServices"));
exports.AppBskyLabelerService = __importStar(require("./types/app/bsky/labeler/service"));
exports.AppBskyNotificationGetUnreadCount = __importStar(require("./types/app/bsky/notification/getUnreadCount"));
exports.AppBskyNotificationListNotifications = __importStar(require("./types/app/bsky/notification/listNotifications"));
exports.AppBskyNotificationPutPreferences = __importStar(require("./types/app/bsky/notification/putPreferences"));
exports.AppBskyNotificationRegisterPush = __importStar(require("./types/app/bsky/notification/registerPush"));
exports.AppBskyNotificationUpdateSeen = __importStar(require("./types/app/bsky/notification/updateSeen"));
exports.AppBskyRichtextFacet = __importStar(require("./types/app/bsky/richtext/facet"));
exports.AppBskyUnspeccedDefs = __importStar(require("./types/app/bsky/unspecced/defs"));
exports.AppBskyUnspeccedGetPopularFeedGenerators = __importStar(require("./types/app/bsky/unspecced/getPopularFeedGenerators"));
exports.AppBskyUnspeccedGetSuggestionsSkeleton = __importStar(require("./types/app/bsky/unspecced/getSuggestionsSkeleton"));
exports.AppBskyUnspeccedGetTaggedSuggestions = __importStar(require("./types/app/bsky/unspecced/getTaggedSuggestions"));
exports.AppBskyUnspeccedSearchActorsSkeleton = __importStar(require("./types/app/bsky/unspecced/searchActorsSkeleton"));
exports.AppBskyUnspeccedSearchPostsSkeleton = __importStar(require("./types/app/bsky/unspecced/searchPostsSkeleton"));
exports.ChatBskyActorDeclaration = __importStar(require("./types/chat/bsky/actor/declaration"));
exports.ChatBskyActorDefs = __importStar(require("./types/chat/bsky/actor/defs"));
exports.ChatBskyActorDeleteAccount = __importStar(require("./types/chat/bsky/actor/deleteAccount"));
exports.ChatBskyActorExportAccountData = __importStar(require("./types/chat/bsky/actor/exportAccountData"));
exports.ChatBskyConvoDefs = __importStar(require("./types/chat/bsky/convo/defs"));
exports.ChatBskyConvoDeleteMessageForSelf = __importStar(require("./types/chat/bsky/convo/deleteMessageForSelf"));
exports.ChatBskyConvoGetConvo = __importStar(require("./types/chat/bsky/convo/getConvo"));
exports.ChatBskyConvoGetConvoForMembers = __importStar(require("./types/chat/bsky/convo/getConvoForMembers"));
exports.ChatBskyConvoGetLog = __importStar(require("./types/chat/bsky/convo/getLog"));
exports.ChatBskyConvoGetMessages = __importStar(require("./types/chat/bsky/convo/getMessages"));
exports.ChatBskyConvoLeaveConvo = __importStar(require("./types/chat/bsky/convo/leaveConvo"));
exports.ChatBskyConvoListConvos = __importStar(require("./types/chat/bsky/convo/listConvos"));
exports.ChatBskyConvoMuteConvo = __importStar(require("./types/chat/bsky/convo/muteConvo"));
exports.ChatBskyConvoSendMessage = __importStar(require("./types/chat/bsky/convo/sendMessage"));
exports.ChatBskyConvoSendMessageBatch = __importStar(require("./types/chat/bsky/convo/sendMessageBatch"));
exports.ChatBskyConvoUnmuteConvo = __importStar(require("./types/chat/bsky/convo/unmuteConvo"));
exports.ChatBskyConvoUpdateRead = __importStar(require("./types/chat/bsky/convo/updateRead"));
exports.ChatBskyModerationGetActorMetadata = __importStar(require("./types/chat/bsky/moderation/getActorMetadata"));
exports.ChatBskyModerationGetMessageContext = __importStar(require("./types/chat/bsky/moderation/getMessageContext"));
exports.ChatBskyModerationUpdateActorAccess = __importStar(require("./types/chat/bsky/moderation/updateActorAccess"));
exports.ToolsOzoneCommunicationCreateTemplate = __importStar(require("./types/tools/ozone/communication/createTemplate"));
exports.ToolsOzoneCommunicationDefs = __importStar(require("./types/tools/ozone/communication/defs"));
exports.ToolsOzoneCommunicationDeleteTemplate = __importStar(require("./types/tools/ozone/communication/deleteTemplate"));
exports.ToolsOzoneCommunicationListTemplates = __importStar(require("./types/tools/ozone/communication/listTemplates"));
exports.ToolsOzoneCommunicationUpdateTemplate = __importStar(require("./types/tools/ozone/communication/updateTemplate"));
exports.ToolsOzoneModerationDefs = __importStar(require("./types/tools/ozone/moderation/defs"));
exports.ToolsOzoneModerationEmitEvent = __importStar(require("./types/tools/ozone/moderation/emitEvent"));
exports.ToolsOzoneModerationGetEvent = __importStar(require("./types/tools/ozone/moderation/getEvent"));
exports.ToolsOzoneModerationGetRecord = __importStar(require("./types/tools/ozone/moderation/getRecord"));
exports.ToolsOzoneModerationGetRepo = __importStar(require("./types/tools/ozone/moderation/getRepo"));
exports.ToolsOzoneModerationQueryEvents = __importStar(require("./types/tools/ozone/moderation/queryEvents"));
exports.ToolsOzoneModerationQueryStatuses = __importStar(require("./types/tools/ozone/moderation/queryStatuses"));
exports.ToolsOzoneModerationSearchRepos = __importStar(require("./types/tools/ozone/moderation/searchRepos"));
exports.ToolsOzoneServerGetConfig = __importStar(require("./types/tools/ozone/server/getConfig"));
exports.ToolsOzoneTeamAddMember = __importStar(require("./types/tools/ozone/team/addMember"));
exports.ToolsOzoneTeamDefs = __importStar(require("./types/tools/ozone/team/defs"));
exports.ToolsOzoneTeamDeleteMember = __importStar(require("./types/tools/ozone/team/deleteMember"));
exports.ToolsOzoneTeamListMembers = __importStar(require("./types/tools/ozone/team/listMembers"));
exports.ToolsOzoneTeamUpdateMember = __importStar(require("./types/tools/ozone/team/updateMember"));
exports.COM_ATPROTO_MODERATION = {
    DefsReasonSpam: 'com.atproto.moderation.defs#reasonSpam',
    DefsReasonViolation: 'com.atproto.moderation.defs#reasonViolation',
    DefsReasonMisleading: 'com.atproto.moderation.defs#reasonMisleading',
    DefsReasonSexual: 'com.atproto.moderation.defs#reasonSexual',
    DefsReasonRude: 'com.atproto.moderation.defs#reasonRude',
    DefsReasonOther: 'com.atproto.moderation.defs#reasonOther',
    DefsReasonAppeal: 'com.atproto.moderation.defs#reasonAppeal',
};
exports.APP_BSKY_FEED = {
    DefsRequestLess: 'app.bsky.feed.defs#requestLess',
    DefsRequestMore: 'app.bsky.feed.defs#requestMore',
    DefsClickthroughItem: 'app.bsky.feed.defs#clickthroughItem',
    DefsClickthroughAuthor: 'app.bsky.feed.defs#clickthroughAuthor',
    DefsClickthroughReposter: 'app.bsky.feed.defs#clickthroughReposter',
    DefsClickthroughEmbed: 'app.bsky.feed.defs#clickthroughEmbed',
    DefsInteractionSeen: 'app.bsky.feed.defs#interactionSeen',
    DefsInteractionLike: 'app.bsky.feed.defs#interactionLike',
    DefsInteractionRepost: 'app.bsky.feed.defs#interactionRepost',
    DefsInteractionReply: 'app.bsky.feed.defs#interactionReply',
    DefsInteractionQuote: 'app.bsky.feed.defs#interactionQuote',
    DefsInteractionShare: 'app.bsky.feed.defs#interactionShare',
};
exports.APP_BSKY_GRAPH = {
    DefsModlist: 'app.bsky.graph.defs#modlist',
    DefsCuratelist: 'app.bsky.graph.defs#curatelist',
    DefsReferencelist: 'app.bsky.graph.defs#referencelist',
};
exports.TOOLS_OZONE_MODERATION = {
    DefsReviewOpen: 'tools.ozone.moderation.defs#reviewOpen',
    DefsReviewEscalated: 'tools.ozone.moderation.defs#reviewEscalated',
    DefsReviewClosed: 'tools.ozone.moderation.defs#reviewClosed',
    DefsReviewNone: 'tools.ozone.moderation.defs#reviewNone',
};
exports.TOOLS_OZONE_TEAM = {
    DefsRoleAdmin: 'tools.ozone.team.defs#roleAdmin',
    DefsRoleModerator: 'tools.ozone.team.defs#roleModerator',
    DefsRoleTriage: 'tools.ozone.team.defs#roleTriage',
};
class AtpBaseClient {
    constructor() {
        Object.defineProperty(this, "xrpc", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: new xrpc_1.Client()
        });
        this.xrpc.addLexicons(lexicons_1.schemas);
    }
    service(serviceUri) {
        return new AtpServiceClient(this, this.xrpc.service(serviceUri));
    }
}
exports.AtpBaseClient = AtpBaseClient;
class AtpServiceClient {
    constructor(baseClient, xrpcService) {
        Object.defineProperty(this, "_baseClient", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "xrpc", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "com", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "app", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "chat", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "tools", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._baseClient = baseClient;
        this.xrpc = xrpcService;
        this.com = new ComNS(this);
        this.app = new AppNS(this);
        this.chat = new ChatNS(this);
        this.tools = new ToolsNS(this);
    }
    setHeader(key, value) {
        this.xrpc.setHeader(key, value);
    }
}
exports.AtpServiceClient = AtpServiceClient;
class ComNS {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "atproto", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
        this.atproto = new ComAtprotoNS(service);
    }
}
exports.ComNS = ComNS;
class ComAtprotoNS {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "admin", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "identity", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "label", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "moderation", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "repo", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "server", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "sync", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "temp", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
        this.admin = new ComAtprotoAdminNS(service);
        this.identity = new ComAtprotoIdentityNS(service);
        this.label = new ComAtprotoLabelNS(service);
        this.moderation = new ComAtprotoModerationNS(service);
        this.repo = new ComAtprotoRepoNS(service);
        this.server = new ComAtprotoServerNS(service);
        this.sync = new ComAtprotoSyncNS(service);
        this.temp = new ComAtprotoTempNS(service);
    }
}
exports.ComAtprotoNS = ComAtprotoNS;
class ComAtprotoAdminNS {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
    }
    deleteAccount(data, opts) {
        return this._service.xrpc
            .call('com.atproto.admin.deleteAccount', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoAdminDeleteAccount.toKnownErr(e);
        });
    }
    disableAccountInvites(data, opts) {
        return this._service.xrpc
            .call('com.atproto.admin.disableAccountInvites', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoAdminDisableAccountInvites.toKnownErr(e);
        });
    }
    disableInviteCodes(data, opts) {
        return this._service.xrpc
            .call('com.atproto.admin.disableInviteCodes', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoAdminDisableInviteCodes.toKnownErr(e);
        });
    }
    enableAccountInvites(data, opts) {
        return this._service.xrpc
            .call('com.atproto.admin.enableAccountInvites', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoAdminEnableAccountInvites.toKnownErr(e);
        });
    }
    getAccountInfo(params, opts) {
        return this._service.xrpc
            .call('com.atproto.admin.getAccountInfo', params, undefined, opts)
            .catch((e) => {
            throw ComAtprotoAdminGetAccountInfo.toKnownErr(e);
        });
    }
    getAccountInfos(params, opts) {
        return this._service.xrpc
            .call('com.atproto.admin.getAccountInfos', params, undefined, opts)
            .catch((e) => {
            throw ComAtprotoAdminGetAccountInfos.toKnownErr(e);
        });
    }
    getInviteCodes(params, opts) {
        return this._service.xrpc
            .call('com.atproto.admin.getInviteCodes', params, undefined, opts)
            .catch((e) => {
            throw ComAtprotoAdminGetInviteCodes.toKnownErr(e);
        });
    }
    getSubjectStatus(params, opts) {
        return this._service.xrpc
            .call('com.atproto.admin.getSubjectStatus', params, undefined, opts)
            .catch((e) => {
            throw ComAtprotoAdminGetSubjectStatus.toKnownErr(e);
        });
    }
    searchAccounts(params, opts) {
        return this._service.xrpc
            .call('com.atproto.admin.searchAccounts', params, undefined, opts)
            .catch((e) => {
            throw ComAtprotoAdminSearchAccounts.toKnownErr(e);
        });
    }
    sendEmail(data, opts) {
        return this._service.xrpc
            .call('com.atproto.admin.sendEmail', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoAdminSendEmail.toKnownErr(e);
        });
    }
    updateAccountEmail(data, opts) {
        return this._service.xrpc
            .call('com.atproto.admin.updateAccountEmail', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoAdminUpdateAccountEmail.toKnownErr(e);
        });
    }
    updateAccountHandle(data, opts) {
        return this._service.xrpc
            .call('com.atproto.admin.updateAccountHandle', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoAdminUpdateAccountHandle.toKnownErr(e);
        });
    }
    updateAccountPassword(data, opts) {
        return this._service.xrpc
            .call('com.atproto.admin.updateAccountPassword', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoAdminUpdateAccountPassword.toKnownErr(e);
        });
    }
    updateSubjectStatus(data, opts) {
        return this._service.xrpc
            .call('com.atproto.admin.updateSubjectStatus', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoAdminUpdateSubjectStatus.toKnownErr(e);
        });
    }
}
exports.ComAtprotoAdminNS = ComAtprotoAdminNS;
class ComAtprotoIdentityNS {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
    }
    getRecommendedDidCredentials(params, opts) {
        return this._service.xrpc
            .call('com.atproto.identity.getRecommendedDidCredentials', params, undefined, opts)
            .catch((e) => {
            throw ComAtprotoIdentityGetRecommendedDidCredentials.toKnownErr(e);
        });
    }
    requestPlcOperationSignature(data, opts) {
        return this._service.xrpc
            .call('com.atproto.identity.requestPlcOperationSignature', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoIdentityRequestPlcOperationSignature.toKnownErr(e);
        });
    }
    resolveHandle(params, opts) {
        return this._service.xrpc
            .call('com.atproto.identity.resolveHandle', params, undefined, opts)
            .catch((e) => {
            throw ComAtprotoIdentityResolveHandle.toKnownErr(e);
        });
    }
    signPlcOperation(data, opts) {
        return this._service.xrpc
            .call('com.atproto.identity.signPlcOperation', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoIdentitySignPlcOperation.toKnownErr(e);
        });
    }
    submitPlcOperation(data, opts) {
        return this._service.xrpc
            .call('com.atproto.identity.submitPlcOperation', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoIdentitySubmitPlcOperation.toKnownErr(e);
        });
    }
    updateHandle(data, opts) {
        return this._service.xrpc
            .call('com.atproto.identity.updateHandle', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoIdentityUpdateHandle.toKnownErr(e);
        });
    }
}
exports.ComAtprotoIdentityNS = ComAtprotoIdentityNS;
class ComAtprotoLabelNS {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
    }
    queryLabels(params, opts) {
        return this._service.xrpc
            .call('com.atproto.label.queryLabels', params, undefined, opts)
            .catch((e) => {
            throw ComAtprotoLabelQueryLabels.toKnownErr(e);
        });
    }
}
exports.ComAtprotoLabelNS = ComAtprotoLabelNS;
class ComAtprotoModerationNS {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
    }
    createReport(data, opts) {
        return this._service.xrpc
            .call('com.atproto.moderation.createReport', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoModerationCreateReport.toKnownErr(e);
        });
    }
}
exports.ComAtprotoModerationNS = ComAtprotoModerationNS;
class ComAtprotoRepoNS {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
    }
    applyWrites(data, opts) {
        return this._service.xrpc
            .call('com.atproto.repo.applyWrites', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoRepoApplyWrites.toKnownErr(e);
        });
    }
    createRecord(data, opts) {
        return this._service.xrpc
            .call('com.atproto.repo.createRecord', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoRepoCreateRecord.toKnownErr(e);
        });
    }
    deleteRecord(data, opts) {
        return this._service.xrpc
            .call('com.atproto.repo.deleteRecord', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoRepoDeleteRecord.toKnownErr(e);
        });
    }
    describeRepo(params, opts) {
        return this._service.xrpc
            .call('com.atproto.repo.describeRepo', params, undefined, opts)
            .catch((e) => {
            throw ComAtprotoRepoDescribeRepo.toKnownErr(e);
        });
    }
    getRecord(params, opts) {
        return this._service.xrpc
            .call('com.atproto.repo.getRecord', params, undefined, opts)
            .catch((e) => {
            throw ComAtprotoRepoGetRecord.toKnownErr(e);
        });
    }
    importRepo(data, opts) {
        return this._service.xrpc
            .call('com.atproto.repo.importRepo', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoRepoImportRepo.toKnownErr(e);
        });
    }
    listMissingBlobs(params, opts) {
        return this._service.xrpc
            .call('com.atproto.repo.listMissingBlobs', params, undefined, opts)
            .catch((e) => {
            throw ComAtprotoRepoListMissingBlobs.toKnownErr(e);
        });
    }
    listRecords(params, opts) {
        return this._service.xrpc
            .call('com.atproto.repo.listRecords', params, undefined, opts)
            .catch((e) => {
            throw ComAtprotoRepoListRecords.toKnownErr(e);
        });
    }
    putRecord(data, opts) {
        return this._service.xrpc
            .call('com.atproto.repo.putRecord', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoRepoPutRecord.toKnownErr(e);
        });
    }
    uploadBlob(data, opts) {
        return this._service.xrpc
            .call('com.atproto.repo.uploadBlob', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoRepoUploadBlob.toKnownErr(e);
        });
    }
}
exports.ComAtprotoRepoNS = ComAtprotoRepoNS;
class ComAtprotoServerNS {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
    }
    activateAccount(data, opts) {
        return this._service.xrpc
            .call('com.atproto.server.activateAccount', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoServerActivateAccount.toKnownErr(e);
        });
    }
    checkAccountStatus(params, opts) {
        return this._service.xrpc
            .call('com.atproto.server.checkAccountStatus', params, undefined, opts)
            .catch((e) => {
            throw ComAtprotoServerCheckAccountStatus.toKnownErr(e);
        });
    }
    confirmEmail(data, opts) {
        return this._service.xrpc
            .call('com.atproto.server.confirmEmail', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoServerConfirmEmail.toKnownErr(e);
        });
    }
    createAccount(data, opts) {
        return this._service.xrpc
            .call('com.atproto.server.createAccount', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoServerCreateAccount.toKnownErr(e);
        });
    }
    createAppPassword(data, opts) {
        return this._service.xrpc
            .call('com.atproto.server.createAppPassword', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoServerCreateAppPassword.toKnownErr(e);
        });
    }
    createInviteCode(data, opts) {
        return this._service.xrpc
            .call('com.atproto.server.createInviteCode', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoServerCreateInviteCode.toKnownErr(e);
        });
    }
    createInviteCodes(data, opts) {
        return this._service.xrpc
            .call('com.atproto.server.createInviteCodes', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoServerCreateInviteCodes.toKnownErr(e);
        });
    }
    createSession(data, opts) {
        return this._service.xrpc
            .call('com.atproto.server.createSession', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoServerCreateSession.toKnownErr(e);
        });
    }
    deactivateAccount(data, opts) {
        return this._service.xrpc
            .call('com.atproto.server.deactivateAccount', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoServerDeactivateAccount.toKnownErr(e);
        });
    }
    deleteAccount(data, opts) {
        return this._service.xrpc
            .call('com.atproto.server.deleteAccount', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoServerDeleteAccount.toKnownErr(e);
        });
    }
    deleteSession(data, opts) {
        return this._service.xrpc
            .call('com.atproto.server.deleteSession', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoServerDeleteSession.toKnownErr(e);
        });
    }
    describeServer(params, opts) {
        return this._service.xrpc
            .call('com.atproto.server.describeServer', params, undefined, opts)
            .catch((e) => {
            throw ComAtprotoServerDescribeServer.toKnownErr(e);
        });
    }
    getAccountInviteCodes(params, opts) {
        return this._service.xrpc
            .call('com.atproto.server.getAccountInviteCodes', params, undefined, opts)
            .catch((e) => {
            throw ComAtprotoServerGetAccountInviteCodes.toKnownErr(e);
        });
    }
    getServiceAuth(params, opts) {
        return this._service.xrpc
            .call('com.atproto.server.getServiceAuth', params, undefined, opts)
            .catch((e) => {
            throw ComAtprotoServerGetServiceAuth.toKnownErr(e);
        });
    }
    getSession(params, opts) {
        return this._service.xrpc
            .call('com.atproto.server.getSession', params, undefined, opts)
            .catch((e) => {
            throw ComAtprotoServerGetSession.toKnownErr(e);
        });
    }
    listAppPasswords(params, opts) {
        return this._service.xrpc
            .call('com.atproto.server.listAppPasswords', params, undefined, opts)
            .catch((e) => {
            throw ComAtprotoServerListAppPasswords.toKnownErr(e);
        });
    }
    refreshSession(data, opts) {
        return this._service.xrpc
            .call('com.atproto.server.refreshSession', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoServerRefreshSession.toKnownErr(e);
        });
    }
    requestAccountDelete(data, opts) {
        return this._service.xrpc
            .call('com.atproto.server.requestAccountDelete', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoServerRequestAccountDelete.toKnownErr(e);
        });
    }
    requestEmailConfirmation(data, opts) {
        return this._service.xrpc
            .call('com.atproto.server.requestEmailConfirmation', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoServerRequestEmailConfirmation.toKnownErr(e);
        });
    }
    requestEmailUpdate(data, opts) {
        return this._service.xrpc
            .call('com.atproto.server.requestEmailUpdate', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoServerRequestEmailUpdate.toKnownErr(e);
        });
    }
    requestPasswordReset(data, opts) {
        return this._service.xrpc
            .call('com.atproto.server.requestPasswordReset', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoServerRequestPasswordReset.toKnownErr(e);
        });
    }
    reserveSigningKey(data, opts) {
        return this._service.xrpc
            .call('com.atproto.server.reserveSigningKey', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoServerReserveSigningKey.toKnownErr(e);
        });
    }
    resetPassword(data, opts) {
        return this._service.xrpc
            .call('com.atproto.server.resetPassword', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoServerResetPassword.toKnownErr(e);
        });
    }
    revokeAppPassword(data, opts) {
        return this._service.xrpc
            .call('com.atproto.server.revokeAppPassword', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoServerRevokeAppPassword.toKnownErr(e);
        });
    }
    updateEmail(data, opts) {
        return this._service.xrpc
            .call('com.atproto.server.updateEmail', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoServerUpdateEmail.toKnownErr(e);
        });
    }
}
exports.ComAtprotoServerNS = ComAtprotoServerNS;
class ComAtprotoSyncNS {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
    }
    getBlob(params, opts) {
        return this._service.xrpc
            .call('com.atproto.sync.getBlob', params, undefined, opts)
            .catch((e) => {
            throw ComAtprotoSyncGetBlob.toKnownErr(e);
        });
    }
    getBlocks(params, opts) {
        return this._service.xrpc
            .call('com.atproto.sync.getBlocks', params, undefined, opts)
            .catch((e) => {
            throw ComAtprotoSyncGetBlocks.toKnownErr(e);
        });
    }
    getCheckout(params, opts) {
        return this._service.xrpc
            .call('com.atproto.sync.getCheckout', params, undefined, opts)
            .catch((e) => {
            throw ComAtprotoSyncGetCheckout.toKnownErr(e);
        });
    }
    getHead(params, opts) {
        return this._service.xrpc
            .call('com.atproto.sync.getHead', params, undefined, opts)
            .catch((e) => {
            throw ComAtprotoSyncGetHead.toKnownErr(e);
        });
    }
    getLatestCommit(params, opts) {
        return this._service.xrpc
            .call('com.atproto.sync.getLatestCommit', params, undefined, opts)
            .catch((e) => {
            throw ComAtprotoSyncGetLatestCommit.toKnownErr(e);
        });
    }
    getRecord(params, opts) {
        return this._service.xrpc
            .call('com.atproto.sync.getRecord', params, undefined, opts)
            .catch((e) => {
            throw ComAtprotoSyncGetRecord.toKnownErr(e);
        });
    }
    getRepo(params, opts) {
        return this._service.xrpc
            .call('com.atproto.sync.getRepo', params, undefined, opts)
            .catch((e) => {
            throw ComAtprotoSyncGetRepo.toKnownErr(e);
        });
    }
    getRepoStatus(params, opts) {
        return this._service.xrpc
            .call('com.atproto.sync.getRepoStatus', params, undefined, opts)
            .catch((e) => {
            throw ComAtprotoSyncGetRepoStatus.toKnownErr(e);
        });
    }
    listBlobs(params, opts) {
        return this._service.xrpc
            .call('com.atproto.sync.listBlobs', params, undefined, opts)
            .catch((e) => {
            throw ComAtprotoSyncListBlobs.toKnownErr(e);
        });
    }
    listRepos(params, opts) {
        return this._service.xrpc
            .call('com.atproto.sync.listRepos', params, undefined, opts)
            .catch((e) => {
            throw ComAtprotoSyncListRepos.toKnownErr(e);
        });
    }
    notifyOfUpdate(data, opts) {
        return this._service.xrpc
            .call('com.atproto.sync.notifyOfUpdate', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoSyncNotifyOfUpdate.toKnownErr(e);
        });
    }
    requestCrawl(data, opts) {
        return this._service.xrpc
            .call('com.atproto.sync.requestCrawl', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoSyncRequestCrawl.toKnownErr(e);
        });
    }
}
exports.ComAtprotoSyncNS = ComAtprotoSyncNS;
class ComAtprotoTempNS {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
    }
    checkSignupQueue(params, opts) {
        return this._service.xrpc
            .call('com.atproto.temp.checkSignupQueue', params, undefined, opts)
            .catch((e) => {
            throw ComAtprotoTempCheckSignupQueue.toKnownErr(e);
        });
    }
    fetchLabels(params, opts) {
        return this._service.xrpc
            .call('com.atproto.temp.fetchLabels', params, undefined, opts)
            .catch((e) => {
            throw ComAtprotoTempFetchLabels.toKnownErr(e);
        });
    }
    requestPhoneVerification(data, opts) {
        return this._service.xrpc
            .call('com.atproto.temp.requestPhoneVerification', opts?.qp, data, opts)
            .catch((e) => {
            throw ComAtprotoTempRequestPhoneVerification.toKnownErr(e);
        });
    }
}
exports.ComAtprotoTempNS = ComAtprotoTempNS;
class AppNS {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "bsky", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
        this.bsky = new AppBskyNS(service);
    }
}
exports.AppNS = AppNS;
class AppBskyNS {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "actor", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "embed", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "feed", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "graph", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "labeler", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "notification", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "richtext", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "unspecced", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
        this.actor = new AppBskyActorNS(service);
        this.embed = new AppBskyEmbedNS(service);
        this.feed = new AppBskyFeedNS(service);
        this.graph = new AppBskyGraphNS(service);
        this.labeler = new AppBskyLabelerNS(service);
        this.notification = new AppBskyNotificationNS(service);
        this.richtext = new AppBskyRichtextNS(service);
        this.unspecced = new AppBskyUnspeccedNS(service);
    }
}
exports.AppBskyNS = AppBskyNS;
class AppBskyActorNS {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "profile", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
        this.profile = new ProfileRecord(service);
    }
    getPreferences(params, opts) {
        return this._service.xrpc
            .call('app.bsky.actor.getPreferences', params, undefined, opts)
            .catch((e) => {
            throw AppBskyActorGetPreferences.toKnownErr(e);
        });
    }
    getProfile(params, opts) {
        return this._service.xrpc
            .call('app.bsky.actor.getProfile', params, undefined, opts)
            .catch((e) => {
            throw AppBskyActorGetProfile.toKnownErr(e);
        });
    }
    getProfiles(params, opts) {
        return this._service.xrpc
            .call('app.bsky.actor.getProfiles', params, undefined, opts)
            .catch((e) => {
            throw AppBskyActorGetProfiles.toKnownErr(e);
        });
    }
    getSuggestions(params, opts) {
        return this._service.xrpc
            .call('app.bsky.actor.getSuggestions', params, undefined, opts)
            .catch((e) => {
            throw AppBskyActorGetSuggestions.toKnownErr(e);
        });
    }
    putPreferences(data, opts) {
        return this._service.xrpc
            .call('app.bsky.actor.putPreferences', opts?.qp, data, opts)
            .catch((e) => {
            throw AppBskyActorPutPreferences.toKnownErr(e);
        });
    }
    searchActors(params, opts) {
        return this._service.xrpc
            .call('app.bsky.actor.searchActors', params, undefined, opts)
            .catch((e) => {
            throw AppBskyActorSearchActors.toKnownErr(e);
        });
    }
    searchActorsTypeahead(params, opts) {
        return this._service.xrpc
            .call('app.bsky.actor.searchActorsTypeahead', params, undefined, opts)
            .catch((e) => {
            throw AppBskyActorSearchActorsTypeahead.toKnownErr(e);
        });
    }
}
exports.AppBskyActorNS = AppBskyActorNS;
class ProfileRecord {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
    }
    async list(params) {
        const res = await this._service.xrpc.call('com.atproto.repo.listRecords', {
            collection: 'app.bsky.actor.profile',
            ...params,
        });
        return res.data;
    }
    async get(params) {
        const res = await this._service.xrpc.call('com.atproto.repo.getRecord', {
            collection: 'app.bsky.actor.profile',
            ...params,
        });
        return res.data;
    }
    async create(params, record, headers) {
        record.$type = 'app.bsky.actor.profile';
        const res = await this._service.xrpc.call('com.atproto.repo.createRecord', undefined, { collection: 'app.bsky.actor.profile', rkey: 'self', ...params, record }, { encoding: 'application/json', headers });
        return res.data;
    }
    async delete(params, headers) {
        await this._service.xrpc.call('com.atproto.repo.deleteRecord', undefined, { collection: 'app.bsky.actor.profile', ...params }, { headers });
    }
}
exports.ProfileRecord = ProfileRecord;
class AppBskyEmbedNS {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
    }
}
exports.AppBskyEmbedNS = AppBskyEmbedNS;
class AppBskyFeedNS {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "generator", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "like", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "post", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "repost", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "threadgate", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
        this.generator = new GeneratorRecord(service);
        this.like = new LikeRecord(service);
        this.post = new PostRecord(service);
        this.repost = new RepostRecord(service);
        this.threadgate = new ThreadgateRecord(service);
    }
    describeFeedGenerator(params, opts) {
        return this._service.xrpc
            .call('app.bsky.feed.describeFeedGenerator', params, undefined, opts)
            .catch((e) => {
            throw AppBskyFeedDescribeFeedGenerator.toKnownErr(e);
        });
    }
    getActorFeeds(params, opts) {
        return this._service.xrpc
            .call('app.bsky.feed.getActorFeeds', params, undefined, opts)
            .catch((e) => {
            throw AppBskyFeedGetActorFeeds.toKnownErr(e);
        });
    }
    getActorLikes(params, opts) {
        return this._service.xrpc
            .call('app.bsky.feed.getActorLikes', params, undefined, opts)
            .catch((e) => {
            throw AppBskyFeedGetActorLikes.toKnownErr(e);
        });
    }
    getAuthorFeed(params, opts) {
        return this._service.xrpc
            .call('app.bsky.feed.getAuthorFeed', params, undefined, opts)
            .catch((e) => {
            throw AppBskyFeedGetAuthorFeed.toKnownErr(e);
        });
    }
    getFeed(params, opts) {
        return this._service.xrpc
            .call('app.bsky.feed.getFeed', params, undefined, opts)
            .catch((e) => {
            throw AppBskyFeedGetFeed.toKnownErr(e);
        });
    }
    getFeedGenerator(params, opts) {
        return this._service.xrpc
            .call('app.bsky.feed.getFeedGenerator', params, undefined, opts)
            .catch((e) => {
            throw AppBskyFeedGetFeedGenerator.toKnownErr(e);
        });
    }
    getFeedGenerators(params, opts) {
        return this._service.xrpc
            .call('app.bsky.feed.getFeedGenerators', params, undefined, opts)
            .catch((e) => {
            throw AppBskyFeedGetFeedGenerators.toKnownErr(e);
        });
    }
    getFeedSkeleton(params, opts) {
        return this._service.xrpc
            .call('app.bsky.feed.getFeedSkeleton', params, undefined, opts)
            .catch((e) => {
            throw AppBskyFeedGetFeedSkeleton.toKnownErr(e);
        });
    }
    getLikes(params, opts) {
        return this._service.xrpc
            .call('app.bsky.feed.getLikes', params, undefined, opts)
            .catch((e) => {
            throw AppBskyFeedGetLikes.toKnownErr(e);
        });
    }
    getListFeed(params, opts) {
        return this._service.xrpc
            .call('app.bsky.feed.getListFeed', params, undefined, opts)
            .catch((e) => {
            throw AppBskyFeedGetListFeed.toKnownErr(e);
        });
    }
    getPostThread(params, opts) {
        return this._service.xrpc
            .call('app.bsky.feed.getPostThread', params, undefined, opts)
            .catch((e) => {
            throw AppBskyFeedGetPostThread.toKnownErr(e);
        });
    }
    getPosts(params, opts) {
        return this._service.xrpc
            .call('app.bsky.feed.getPosts', params, undefined, opts)
            .catch((e) => {
            throw AppBskyFeedGetPosts.toKnownErr(e);
        });
    }
    getRepostedBy(params, opts) {
        return this._service.xrpc
            .call('app.bsky.feed.getRepostedBy', params, undefined, opts)
            .catch((e) => {
            throw AppBskyFeedGetRepostedBy.toKnownErr(e);
        });
    }
    getSuggestedFeeds(params, opts) {
        return this._service.xrpc
            .call('app.bsky.feed.getSuggestedFeeds', params, undefined, opts)
            .catch((e) => {
            throw AppBskyFeedGetSuggestedFeeds.toKnownErr(e);
        });
    }
    getTimeline(params, opts) {
        return this._service.xrpc
            .call('app.bsky.feed.getTimeline', params, undefined, opts)
            .catch((e) => {
            throw AppBskyFeedGetTimeline.toKnownErr(e);
        });
    }
    searchPosts(params, opts) {
        return this._service.xrpc
            .call('app.bsky.feed.searchPosts', params, undefined, opts)
            .catch((e) => {
            throw AppBskyFeedSearchPosts.toKnownErr(e);
        });
    }
    sendInteractions(data, opts) {
        return this._service.xrpc
            .call('app.bsky.feed.sendInteractions', opts?.qp, data, opts)
            .catch((e) => {
            throw AppBskyFeedSendInteractions.toKnownErr(e);
        });
    }
}
exports.AppBskyFeedNS = AppBskyFeedNS;
class GeneratorRecord {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
    }
    async list(params) {
        const res = await this._service.xrpc.call('com.atproto.repo.listRecords', {
            collection: 'app.bsky.feed.generator',
            ...params,
        });
        return res.data;
    }
    async get(params) {
        const res = await this._service.xrpc.call('com.atproto.repo.getRecord', {
            collection: 'app.bsky.feed.generator',
            ...params,
        });
        return res.data;
    }
    async create(params, record, headers) {
        record.$type = 'app.bsky.feed.generator';
        const res = await this._service.xrpc.call('com.atproto.repo.createRecord', undefined, { collection: 'app.bsky.feed.generator', ...params, record }, { encoding: 'application/json', headers });
        return res.data;
    }
    async delete(params, headers) {
        await this._service.xrpc.call('com.atproto.repo.deleteRecord', undefined, { collection: 'app.bsky.feed.generator', ...params }, { headers });
    }
}
exports.GeneratorRecord = GeneratorRecord;
class LikeRecord {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
    }
    async list(params) {
        const res = await this._service.xrpc.call('com.atproto.repo.listRecords', {
            collection: 'app.bsky.feed.like',
            ...params,
        });
        return res.data;
    }
    async get(params) {
        const res = await this._service.xrpc.call('com.atproto.repo.getRecord', {
            collection: 'app.bsky.feed.like',
            ...params,
        });
        return res.data;
    }
    async create(params, record, headers) {
        record.$type = 'app.bsky.feed.like';
        const res = await this._service.xrpc.call('com.atproto.repo.createRecord', undefined, { collection: 'app.bsky.feed.like', ...params, record }, { encoding: 'application/json', headers });
        return res.data;
    }
    async delete(params, headers) {
        await this._service.xrpc.call('com.atproto.repo.deleteRecord', undefined, { collection: 'app.bsky.feed.like', ...params }, { headers });
    }
}
exports.LikeRecord = LikeRecord;
class PostRecord {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
    }
    async list(params) {
        const res = await this._service.xrpc.call('com.atproto.repo.listRecords', {
            collection: 'app.bsky.feed.post',
            ...params,
        });
        return res.data;
    }
    async get(params) {
        const res = await this._service.xrpc.call('com.atproto.repo.getRecord', {
            collection: 'app.bsky.feed.post',
            ...params,
        });
        return res.data;
    }
    async create(params, record, headers) {
        record.$type = 'app.bsky.feed.post';
        const res = await this._service.xrpc.call('com.atproto.repo.createRecord', undefined, { collection: 'app.bsky.feed.post', ...params, record }, { encoding: 'application/json', headers });
        return res.data;
    }
    async delete(params, headers) {
        await this._service.xrpc.call('com.atproto.repo.deleteRecord', undefined, { collection: 'app.bsky.feed.post', ...params }, { headers });
    }
}
exports.PostRecord = PostRecord;
class RepostRecord {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
    }
    async list(params) {
        const res = await this._service.xrpc.call('com.atproto.repo.listRecords', {
            collection: 'app.bsky.feed.repost',
            ...params,
        });
        return res.data;
    }
    async get(params) {
        const res = await this._service.xrpc.call('com.atproto.repo.getRecord', {
            collection: 'app.bsky.feed.repost',
            ...params,
        });
        return res.data;
    }
    async create(params, record, headers) {
        record.$type = 'app.bsky.feed.repost';
        const res = await this._service.xrpc.call('com.atproto.repo.createRecord', undefined, { collection: 'app.bsky.feed.repost', ...params, record }, { encoding: 'application/json', headers });
        return res.data;
    }
    async delete(params, headers) {
        await this._service.xrpc.call('com.atproto.repo.deleteRecord', undefined, { collection: 'app.bsky.feed.repost', ...params }, { headers });
    }
}
exports.RepostRecord = RepostRecord;
class ThreadgateRecord {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
    }
    async list(params) {
        const res = await this._service.xrpc.call('com.atproto.repo.listRecords', {
            collection: 'app.bsky.feed.threadgate',
            ...params,
        });
        return res.data;
    }
    async get(params) {
        const res = await this._service.xrpc.call('com.atproto.repo.getRecord', {
            collection: 'app.bsky.feed.threadgate',
            ...params,
        });
        return res.data;
    }
    async create(params, record, headers) {
        record.$type = 'app.bsky.feed.threadgate';
        const res = await this._service.xrpc.call('com.atproto.repo.createRecord', undefined, { collection: 'app.bsky.feed.threadgate', ...params, record }, { encoding: 'application/json', headers });
        return res.data;
    }
    async delete(params, headers) {
        await this._service.xrpc.call('com.atproto.repo.deleteRecord', undefined, { collection: 'app.bsky.feed.threadgate', ...params }, { headers });
    }
}
exports.ThreadgateRecord = ThreadgateRecord;
class AppBskyGraphNS {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "block", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "follow", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "list", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "listblock", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "listitem", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "starterpack", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
        this.block = new BlockRecord(service);
        this.follow = new FollowRecord(service);
        this.list = new ListRecord(service);
        this.listblock = new ListblockRecord(service);
        this.listitem = new ListitemRecord(service);
        this.starterpack = new StarterpackRecord(service);
    }
    getActorStarterPacks(params, opts) {
        return this._service.xrpc
            .call('app.bsky.graph.getActorStarterPacks', params, undefined, opts)
            .catch((e) => {
            throw AppBskyGraphGetActorStarterPacks.toKnownErr(e);
        });
    }
    getBlocks(params, opts) {
        return this._service.xrpc
            .call('app.bsky.graph.getBlocks', params, undefined, opts)
            .catch((e) => {
            throw AppBskyGraphGetBlocks.toKnownErr(e);
        });
    }
    getFollowers(params, opts) {
        return this._service.xrpc
            .call('app.bsky.graph.getFollowers', params, undefined, opts)
            .catch((e) => {
            throw AppBskyGraphGetFollowers.toKnownErr(e);
        });
    }
    getFollows(params, opts) {
        return this._service.xrpc
            .call('app.bsky.graph.getFollows', params, undefined, opts)
            .catch((e) => {
            throw AppBskyGraphGetFollows.toKnownErr(e);
        });
    }
    getKnownFollowers(params, opts) {
        return this._service.xrpc
            .call('app.bsky.graph.getKnownFollowers', params, undefined, opts)
            .catch((e) => {
            throw AppBskyGraphGetKnownFollowers.toKnownErr(e);
        });
    }
    getList(params, opts) {
        return this._service.xrpc
            .call('app.bsky.graph.getList', params, undefined, opts)
            .catch((e) => {
            throw AppBskyGraphGetList.toKnownErr(e);
        });
    }
    getListBlocks(params, opts) {
        return this._service.xrpc
            .call('app.bsky.graph.getListBlocks', params, undefined, opts)
            .catch((e) => {
            throw AppBskyGraphGetListBlocks.toKnownErr(e);
        });
    }
    getListMutes(params, opts) {
        return this._service.xrpc
            .call('app.bsky.graph.getListMutes', params, undefined, opts)
            .catch((e) => {
            throw AppBskyGraphGetListMutes.toKnownErr(e);
        });
    }
    getLists(params, opts) {
        return this._service.xrpc
            .call('app.bsky.graph.getLists', params, undefined, opts)
            .catch((e) => {
            throw AppBskyGraphGetLists.toKnownErr(e);
        });
    }
    getMutes(params, opts) {
        return this._service.xrpc
            .call('app.bsky.graph.getMutes', params, undefined, opts)
            .catch((e) => {
            throw AppBskyGraphGetMutes.toKnownErr(e);
        });
    }
    getRelationships(params, opts) {
        return this._service.xrpc
            .call('app.bsky.graph.getRelationships', params, undefined, opts)
            .catch((e) => {
            throw AppBskyGraphGetRelationships.toKnownErr(e);
        });
    }
    getStarterPack(params, opts) {
        return this._service.xrpc
            .call('app.bsky.graph.getStarterPack', params, undefined, opts)
            .catch((e) => {
            throw AppBskyGraphGetStarterPack.toKnownErr(e);
        });
    }
    getStarterPacks(params, opts) {
        return this._service.xrpc
            .call('app.bsky.graph.getStarterPacks', params, undefined, opts)
            .catch((e) => {
            throw AppBskyGraphGetStarterPacks.toKnownErr(e);
        });
    }
    getSuggestedFollowsByActor(params, opts) {
        return this._service.xrpc
            .call('app.bsky.graph.getSuggestedFollowsByActor', params, undefined, opts)
            .catch((e) => {
            throw AppBskyGraphGetSuggestedFollowsByActor.toKnownErr(e);
        });
    }
    muteActor(data, opts) {
        return this._service.xrpc
            .call('app.bsky.graph.muteActor', opts?.qp, data, opts)
            .catch((e) => {
            throw AppBskyGraphMuteActor.toKnownErr(e);
        });
    }
    muteActorList(data, opts) {
        return this._service.xrpc
            .call('app.bsky.graph.muteActorList', opts?.qp, data, opts)
            .catch((e) => {
            throw AppBskyGraphMuteActorList.toKnownErr(e);
        });
    }
    muteThread(data, opts) {
        return this._service.xrpc
            .call('app.bsky.graph.muteThread', opts?.qp, data, opts)
            .catch((e) => {
            throw AppBskyGraphMuteThread.toKnownErr(e);
        });
    }
    unmuteActor(data, opts) {
        return this._service.xrpc
            .call('app.bsky.graph.unmuteActor', opts?.qp, data, opts)
            .catch((e) => {
            throw AppBskyGraphUnmuteActor.toKnownErr(e);
        });
    }
    unmuteActorList(data, opts) {
        return this._service.xrpc
            .call('app.bsky.graph.unmuteActorList', opts?.qp, data, opts)
            .catch((e) => {
            throw AppBskyGraphUnmuteActorList.toKnownErr(e);
        });
    }
    unmuteThread(data, opts) {
        return this._service.xrpc
            .call('app.bsky.graph.unmuteThread', opts?.qp, data, opts)
            .catch((e) => {
            throw AppBskyGraphUnmuteThread.toKnownErr(e);
        });
    }
}
exports.AppBskyGraphNS = AppBskyGraphNS;
class BlockRecord {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
    }
    async list(params) {
        const res = await this._service.xrpc.call('com.atproto.repo.listRecords', {
            collection: 'app.bsky.graph.block',
            ...params,
        });
        return res.data;
    }
    async get(params) {
        const res = await this._service.xrpc.call('com.atproto.repo.getRecord', {
            collection: 'app.bsky.graph.block',
            ...params,
        });
        return res.data;
    }
    async create(params, record, headers) {
        record.$type = 'app.bsky.graph.block';
        const res = await this._service.xrpc.call('com.atproto.repo.createRecord', undefined, { collection: 'app.bsky.graph.block', ...params, record }, { encoding: 'application/json', headers });
        return res.data;
    }
    async delete(params, headers) {
        await this._service.xrpc.call('com.atproto.repo.deleteRecord', undefined, { collection: 'app.bsky.graph.block', ...params }, { headers });
    }
}
exports.BlockRecord = BlockRecord;
class FollowRecord {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
    }
    async list(params) {
        const res = await this._service.xrpc.call('com.atproto.repo.listRecords', {
            collection: 'app.bsky.graph.follow',
            ...params,
        });
        return res.data;
    }
    async get(params) {
        const res = await this._service.xrpc.call('com.atproto.repo.getRecord', {
            collection: 'app.bsky.graph.follow',
            ...params,
        });
        return res.data;
    }
    async create(params, record, headers) {
        record.$type = 'app.bsky.graph.follow';
        const res = await this._service.xrpc.call('com.atproto.repo.createRecord', undefined, { collection: 'app.bsky.graph.follow', ...params, record }, { encoding: 'application/json', headers });
        return res.data;
    }
    async delete(params, headers) {
        await this._service.xrpc.call('com.atproto.repo.deleteRecord', undefined, { collection: 'app.bsky.graph.follow', ...params }, { headers });
    }
}
exports.FollowRecord = FollowRecord;
class ListRecord {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
    }
    async list(params) {
        const res = await this._service.xrpc.call('com.atproto.repo.listRecords', {
            collection: 'app.bsky.graph.list',
            ...params,
        });
        return res.data;
    }
    async get(params) {
        const res = await this._service.xrpc.call('com.atproto.repo.getRecord', {
            collection: 'app.bsky.graph.list',
            ...params,
        });
        return res.data;
    }
    async create(params, record, headers) {
        record.$type = 'app.bsky.graph.list';
        const res = await this._service.xrpc.call('com.atproto.repo.createRecord', undefined, { collection: 'app.bsky.graph.list', ...params, record }, { encoding: 'application/json', headers });
        return res.data;
    }
    async delete(params, headers) {
        await this._service.xrpc.call('com.atproto.repo.deleteRecord', undefined, { collection: 'app.bsky.graph.list', ...params }, { headers });
    }
}
exports.ListRecord = ListRecord;
class ListblockRecord {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
    }
    async list(params) {
        const res = await this._service.xrpc.call('com.atproto.repo.listRecords', {
            collection: 'app.bsky.graph.listblock',
            ...params,
        });
        return res.data;
    }
    async get(params) {
        const res = await this._service.xrpc.call('com.atproto.repo.getRecord', {
            collection: 'app.bsky.graph.listblock',
            ...params,
        });
        return res.data;
    }
    async create(params, record, headers) {
        record.$type = 'app.bsky.graph.listblock';
        const res = await this._service.xrpc.call('com.atproto.repo.createRecord', undefined, { collection: 'app.bsky.graph.listblock', ...params, record }, { encoding: 'application/json', headers });
        return res.data;
    }
    async delete(params, headers) {
        await this._service.xrpc.call('com.atproto.repo.deleteRecord', undefined, { collection: 'app.bsky.graph.listblock', ...params }, { headers });
    }
}
exports.ListblockRecord = ListblockRecord;
class ListitemRecord {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
    }
    async list(params) {
        const res = await this._service.xrpc.call('com.atproto.repo.listRecords', {
            collection: 'app.bsky.graph.listitem',
            ...params,
        });
        return res.data;
    }
    async get(params) {
        const res = await this._service.xrpc.call('com.atproto.repo.getRecord', {
            collection: 'app.bsky.graph.listitem',
            ...params,
        });
        return res.data;
    }
    async create(params, record, headers) {
        record.$type = 'app.bsky.graph.listitem';
        const res = await this._service.xrpc.call('com.atproto.repo.createRecord', undefined, { collection: 'app.bsky.graph.listitem', ...params, record }, { encoding: 'application/json', headers });
        return res.data;
    }
    async delete(params, headers) {
        await this._service.xrpc.call('com.atproto.repo.deleteRecord', undefined, { collection: 'app.bsky.graph.listitem', ...params }, { headers });
    }
}
exports.ListitemRecord = ListitemRecord;
class StarterpackRecord {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
    }
    async list(params) {
        const res = await this._service.xrpc.call('com.atproto.repo.listRecords', {
            collection: 'app.bsky.graph.starterpack',
            ...params,
        });
        return res.data;
    }
    async get(params) {
        const res = await this._service.xrpc.call('com.atproto.repo.getRecord', {
            collection: 'app.bsky.graph.starterpack',
            ...params,
        });
        return res.data;
    }
    async create(params, record, headers) {
        record.$type = 'app.bsky.graph.starterpack';
        const res = await this._service.xrpc.call('com.atproto.repo.createRecord', undefined, { collection: 'app.bsky.graph.starterpack', ...params, record }, { encoding: 'application/json', headers });
        return res.data;
    }
    async delete(params, headers) {
        await this._service.xrpc.call('com.atproto.repo.deleteRecord', undefined, { collection: 'app.bsky.graph.starterpack', ...params }, { headers });
    }
}
exports.StarterpackRecord = StarterpackRecord;
class AppBskyLabelerNS {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
        this.service = new ServiceRecord(service);
    }
    getServices(params, opts) {
        return this._service.xrpc
            .call('app.bsky.labeler.getServices', params, undefined, opts)
            .catch((e) => {
            throw AppBskyLabelerGetServices.toKnownErr(e);
        });
    }
}
exports.AppBskyLabelerNS = AppBskyLabelerNS;
class ServiceRecord {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
    }
    async list(params) {
        const res = await this._service.xrpc.call('com.atproto.repo.listRecords', {
            collection: 'app.bsky.labeler.service',
            ...params,
        });
        return res.data;
    }
    async get(params) {
        const res = await this._service.xrpc.call('com.atproto.repo.getRecord', {
            collection: 'app.bsky.labeler.service',
            ...params,
        });
        return res.data;
    }
    async create(params, record, headers) {
        record.$type = 'app.bsky.labeler.service';
        const res = await this._service.xrpc.call('com.atproto.repo.createRecord', undefined, {
            collection: 'app.bsky.labeler.service',
            rkey: 'self',
            ...params,
            record,
        }, { encoding: 'application/json', headers });
        return res.data;
    }
    async delete(params, headers) {
        await this._service.xrpc.call('com.atproto.repo.deleteRecord', undefined, { collection: 'app.bsky.labeler.service', ...params }, { headers });
    }
}
exports.ServiceRecord = ServiceRecord;
class AppBskyNotificationNS {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
    }
    getUnreadCount(params, opts) {
        return this._service.xrpc
            .call('app.bsky.notification.getUnreadCount', params, undefined, opts)
            .catch((e) => {
            throw AppBskyNotificationGetUnreadCount.toKnownErr(e);
        });
    }
    listNotifications(params, opts) {
        return this._service.xrpc
            .call('app.bsky.notification.listNotifications', params, undefined, opts)
            .catch((e) => {
            throw AppBskyNotificationListNotifications.toKnownErr(e);
        });
    }
    putPreferences(data, opts) {
        return this._service.xrpc
            .call('app.bsky.notification.putPreferences', opts?.qp, data, opts)
            .catch((e) => {
            throw AppBskyNotificationPutPreferences.toKnownErr(e);
        });
    }
    registerPush(data, opts) {
        return this._service.xrpc
            .call('app.bsky.notification.registerPush', opts?.qp, data, opts)
            .catch((e) => {
            throw AppBskyNotificationRegisterPush.toKnownErr(e);
        });
    }
    updateSeen(data, opts) {
        return this._service.xrpc
            .call('app.bsky.notification.updateSeen', opts?.qp, data, opts)
            .catch((e) => {
            throw AppBskyNotificationUpdateSeen.toKnownErr(e);
        });
    }
}
exports.AppBskyNotificationNS = AppBskyNotificationNS;
class AppBskyRichtextNS {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
    }
}
exports.AppBskyRichtextNS = AppBskyRichtextNS;
class AppBskyUnspeccedNS {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
    }
    getPopularFeedGenerators(params, opts) {
        return this._service.xrpc
            .call('app.bsky.unspecced.getPopularFeedGenerators', params, undefined, opts)
            .catch((e) => {
            throw AppBskyUnspeccedGetPopularFeedGenerators.toKnownErr(e);
        });
    }
    getSuggestionsSkeleton(params, opts) {
        return this._service.xrpc
            .call('app.bsky.unspecced.getSuggestionsSkeleton', params, undefined, opts)
            .catch((e) => {
            throw AppBskyUnspeccedGetSuggestionsSkeleton.toKnownErr(e);
        });
    }
    getTaggedSuggestions(params, opts) {
        return this._service.xrpc
            .call('app.bsky.unspecced.getTaggedSuggestions', params, undefined, opts)
            .catch((e) => {
            throw AppBskyUnspeccedGetTaggedSuggestions.toKnownErr(e);
        });
    }
    searchActorsSkeleton(params, opts) {
        return this._service.xrpc
            .call('app.bsky.unspecced.searchActorsSkeleton', params, undefined, opts)
            .catch((e) => {
            throw AppBskyUnspeccedSearchActorsSkeleton.toKnownErr(e);
        });
    }
    searchPostsSkeleton(params, opts) {
        return this._service.xrpc
            .call('app.bsky.unspecced.searchPostsSkeleton', params, undefined, opts)
            .catch((e) => {
            throw AppBskyUnspeccedSearchPostsSkeleton.toKnownErr(e);
        });
    }
}
exports.AppBskyUnspeccedNS = AppBskyUnspeccedNS;
class ChatNS {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "bsky", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
        this.bsky = new ChatBskyNS(service);
    }
}
exports.ChatNS = ChatNS;
class ChatBskyNS {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "actor", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "convo", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "moderation", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
        this.actor = new ChatBskyActorNS(service);
        this.convo = new ChatBskyConvoNS(service);
        this.moderation = new ChatBskyModerationNS(service);
    }
}
exports.ChatBskyNS = ChatBskyNS;
class ChatBskyActorNS {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "declaration", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
        this.declaration = new DeclarationRecord(service);
    }
    deleteAccount(data, opts) {
        return this._service.xrpc
            .call('chat.bsky.actor.deleteAccount', opts?.qp, data, opts)
            .catch((e) => {
            throw ChatBskyActorDeleteAccount.toKnownErr(e);
        });
    }
    exportAccountData(params, opts) {
        return this._service.xrpc
            .call('chat.bsky.actor.exportAccountData', params, undefined, opts)
            .catch((e) => {
            throw ChatBskyActorExportAccountData.toKnownErr(e);
        });
    }
}
exports.ChatBskyActorNS = ChatBskyActorNS;
class DeclarationRecord {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
    }
    async list(params) {
        const res = await this._service.xrpc.call('com.atproto.repo.listRecords', {
            collection: 'chat.bsky.actor.declaration',
            ...params,
        });
        return res.data;
    }
    async get(params) {
        const res = await this._service.xrpc.call('com.atproto.repo.getRecord', {
            collection: 'chat.bsky.actor.declaration',
            ...params,
        });
        return res.data;
    }
    async create(params, record, headers) {
        record.$type = 'chat.bsky.actor.declaration';
        const res = await this._service.xrpc.call('com.atproto.repo.createRecord', undefined, {
            collection: 'chat.bsky.actor.declaration',
            rkey: 'self',
            ...params,
            record,
        }, { encoding: 'application/json', headers });
        return res.data;
    }
    async delete(params, headers) {
        await this._service.xrpc.call('com.atproto.repo.deleteRecord', undefined, { collection: 'chat.bsky.actor.declaration', ...params }, { headers });
    }
}
exports.DeclarationRecord = DeclarationRecord;
class ChatBskyConvoNS {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
    }
    deleteMessageForSelf(data, opts) {
        return this._service.xrpc
            .call('chat.bsky.convo.deleteMessageForSelf', opts?.qp, data, opts)
            .catch((e) => {
            throw ChatBskyConvoDeleteMessageForSelf.toKnownErr(e);
        });
    }
    getConvo(params, opts) {
        return this._service.xrpc
            .call('chat.bsky.convo.getConvo', params, undefined, opts)
            .catch((e) => {
            throw ChatBskyConvoGetConvo.toKnownErr(e);
        });
    }
    getConvoForMembers(params, opts) {
        return this._service.xrpc
            .call('chat.bsky.convo.getConvoForMembers', params, undefined, opts)
            .catch((e) => {
            throw ChatBskyConvoGetConvoForMembers.toKnownErr(e);
        });
    }
    getLog(params, opts) {
        return this._service.xrpc
            .call('chat.bsky.convo.getLog', params, undefined, opts)
            .catch((e) => {
            throw ChatBskyConvoGetLog.toKnownErr(e);
        });
    }
    getMessages(params, opts) {
        return this._service.xrpc
            .call('chat.bsky.convo.getMessages', params, undefined, opts)
            .catch((e) => {
            throw ChatBskyConvoGetMessages.toKnownErr(e);
        });
    }
    leaveConvo(data, opts) {
        return this._service.xrpc
            .call('chat.bsky.convo.leaveConvo', opts?.qp, data, opts)
            .catch((e) => {
            throw ChatBskyConvoLeaveConvo.toKnownErr(e);
        });
    }
    listConvos(params, opts) {
        return this._service.xrpc
            .call('chat.bsky.convo.listConvos', params, undefined, opts)
            .catch((e) => {
            throw ChatBskyConvoListConvos.toKnownErr(e);
        });
    }
    muteConvo(data, opts) {
        return this._service.xrpc
            .call('chat.bsky.convo.muteConvo', opts?.qp, data, opts)
            .catch((e) => {
            throw ChatBskyConvoMuteConvo.toKnownErr(e);
        });
    }
    sendMessage(data, opts) {
        return this._service.xrpc
            .call('chat.bsky.convo.sendMessage', opts?.qp, data, opts)
            .catch((e) => {
            throw ChatBskyConvoSendMessage.toKnownErr(e);
        });
    }
    sendMessageBatch(data, opts) {
        return this._service.xrpc
            .call('chat.bsky.convo.sendMessageBatch', opts?.qp, data, opts)
            .catch((e) => {
            throw ChatBskyConvoSendMessageBatch.toKnownErr(e);
        });
    }
    unmuteConvo(data, opts) {
        return this._service.xrpc
            .call('chat.bsky.convo.unmuteConvo', opts?.qp, data, opts)
            .catch((e) => {
            throw ChatBskyConvoUnmuteConvo.toKnownErr(e);
        });
    }
    updateRead(data, opts) {
        return this._service.xrpc
            .call('chat.bsky.convo.updateRead', opts?.qp, data, opts)
            .catch((e) => {
            throw ChatBskyConvoUpdateRead.toKnownErr(e);
        });
    }
}
exports.ChatBskyConvoNS = ChatBskyConvoNS;
class ChatBskyModerationNS {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
    }
    getActorMetadata(params, opts) {
        return this._service.xrpc
            .call('chat.bsky.moderation.getActorMetadata', params, undefined, opts)
            .catch((e) => {
            throw ChatBskyModerationGetActorMetadata.toKnownErr(e);
        });
    }
    getMessageContext(params, opts) {
        return this._service.xrpc
            .call('chat.bsky.moderation.getMessageContext', params, undefined, opts)
            .catch((e) => {
            throw ChatBskyModerationGetMessageContext.toKnownErr(e);
        });
    }
    updateActorAccess(data, opts) {
        return this._service.xrpc
            .call('chat.bsky.moderation.updateActorAccess', opts?.qp, data, opts)
            .catch((e) => {
            throw ChatBskyModerationUpdateActorAccess.toKnownErr(e);
        });
    }
}
exports.ChatBskyModerationNS = ChatBskyModerationNS;
class ToolsNS {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "ozone", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
        this.ozone = new ToolsOzoneNS(service);
    }
}
exports.ToolsNS = ToolsNS;
class ToolsOzoneNS {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "communication", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "moderation", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "server", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "team", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
        this.communication = new ToolsOzoneCommunicationNS(service);
        this.moderation = new ToolsOzoneModerationNS(service);
        this.server = new ToolsOzoneServerNS(service);
        this.team = new ToolsOzoneTeamNS(service);
    }
}
exports.ToolsOzoneNS = ToolsOzoneNS;
class ToolsOzoneCommunicationNS {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
    }
    createTemplate(data, opts) {
        return this._service.xrpc
            .call('tools.ozone.communication.createTemplate', opts?.qp, data, opts)
            .catch((e) => {
            throw ToolsOzoneCommunicationCreateTemplate.toKnownErr(e);
        });
    }
    deleteTemplate(data, opts) {
        return this._service.xrpc
            .call('tools.ozone.communication.deleteTemplate', opts?.qp, data, opts)
            .catch((e) => {
            throw ToolsOzoneCommunicationDeleteTemplate.toKnownErr(e);
        });
    }
    listTemplates(params, opts) {
        return this._service.xrpc
            .call('tools.ozone.communication.listTemplates', params, undefined, opts)
            .catch((e) => {
            throw ToolsOzoneCommunicationListTemplates.toKnownErr(e);
        });
    }
    updateTemplate(data, opts) {
        return this._service.xrpc
            .call('tools.ozone.communication.updateTemplate', opts?.qp, data, opts)
            .catch((e) => {
            throw ToolsOzoneCommunicationUpdateTemplate.toKnownErr(e);
        });
    }
}
exports.ToolsOzoneCommunicationNS = ToolsOzoneCommunicationNS;
class ToolsOzoneModerationNS {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
    }
    emitEvent(data, opts) {
        return this._service.xrpc
            .call('tools.ozone.moderation.emitEvent', opts?.qp, data, opts)
            .catch((e) => {
            throw ToolsOzoneModerationEmitEvent.toKnownErr(e);
        });
    }
    getEvent(params, opts) {
        return this._service.xrpc
            .call('tools.ozone.moderation.getEvent', params, undefined, opts)
            .catch((e) => {
            throw ToolsOzoneModerationGetEvent.toKnownErr(e);
        });
    }
    getRecord(params, opts) {
        return this._service.xrpc
            .call('tools.ozone.moderation.getRecord', params, undefined, opts)
            .catch((e) => {
            throw ToolsOzoneModerationGetRecord.toKnownErr(e);
        });
    }
    getRepo(params, opts) {
        return this._service.xrpc
            .call('tools.ozone.moderation.getRepo', params, undefined, opts)
            .catch((e) => {
            throw ToolsOzoneModerationGetRepo.toKnownErr(e);
        });
    }
    queryEvents(params, opts) {
        return this._service.xrpc
            .call('tools.ozone.moderation.queryEvents', params, undefined, opts)
            .catch((e) => {
            throw ToolsOzoneModerationQueryEvents.toKnownErr(e);
        });
    }
    queryStatuses(params, opts) {
        return this._service.xrpc
            .call('tools.ozone.moderation.queryStatuses', params, undefined, opts)
            .catch((e) => {
            throw ToolsOzoneModerationQueryStatuses.toKnownErr(e);
        });
    }
    searchRepos(params, opts) {
        return this._service.xrpc
            .call('tools.ozone.moderation.searchRepos', params, undefined, opts)
            .catch((e) => {
            throw ToolsOzoneModerationSearchRepos.toKnownErr(e);
        });
    }
}
exports.ToolsOzoneModerationNS = ToolsOzoneModerationNS;
class ToolsOzoneServerNS {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
    }
    getConfig(params, opts) {
        return this._service.xrpc
            .call('tools.ozone.server.getConfig', params, undefined, opts)
            .catch((e) => {
            throw ToolsOzoneServerGetConfig.toKnownErr(e);
        });
    }
}
exports.ToolsOzoneServerNS = ToolsOzoneServerNS;
class ToolsOzoneTeamNS {
    constructor(service) {
        Object.defineProperty(this, "_service", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._service = service;
    }
    addMember(data, opts) {
        return this._service.xrpc
            .call('tools.ozone.team.addMember', opts?.qp, data, opts)
            .catch((e) => {
            throw ToolsOzoneTeamAddMember.toKnownErr(e);
        });
    }
    deleteMember(data, opts) {
        return this._service.xrpc
            .call('tools.ozone.team.deleteMember', opts?.qp, data, opts)
            .catch((e) => {
            throw ToolsOzoneTeamDeleteMember.toKnownErr(e);
        });
    }
    listMembers(params, opts) {
        return this._service.xrpc
            .call('tools.ozone.team.listMembers', params, undefined, opts)
            .catch((e) => {
            throw ToolsOzoneTeamListMembers.toKnownErr(e);
        });
    }
    updateMember(data, opts) {
        return this._service.xrpc
            .call('tools.ozone.team.updateMember', opts?.qp, data, opts)
            .catch((e) => {
            throw ToolsOzoneTeamUpdateMember.toKnownErr(e);
        });
    }
}
exports.ToolsOzoneTeamNS = ToolsOzoneTeamNS;
